{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Phylogenetic Tree"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Creating a distance tree\n",
    "We will use ```phylip``` to create a distance tree from the p53 protein alignment that we created eariler using ```muscle```. We can use a script to convert an existing aligment into ```phylip``` format (I wrote such a script called ```fasta2phylip.pl```) or we can realign the sequences using ```muscle```, but this time we can directly output in ```phylip``` format. We will use the following program from the ```phylip``` package:\n",
    "\n",
    "1. ```protdist``` creates a distance matrix from an alignment\n",
    "2.\t```neighbor``` will create a neighbor-joining tree\n",
    "\n",
    "Remember, all programs in phylip will take an ```infile``` and will create an ```outfile```.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Creating the alignment \n",
    "Copy the fasta file containing the p53 sequences into current directory. ```muscle``` can output two different ```phylip``` formats. We will use \"interleaved\". Note, we could have output several formats from ```muscle``` simultaneously during alignment, like this:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "engine": "bash",
    "eval": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "MUSCLE v3.8.31 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "p53 1 seqs, max length 393, avg  length 393\n",
      ">sp|P04637|P53_HUMAN Cellular tumor antigen p53 OS=Homo sapiens OX=9606 GN=TP53 PE=1 SV=4\n",
      "MEEPQSDPSVEPPLSQETFSDLWKLLPENNVLSPLPSQAMDDLMLSPDDIEQWFTEDPGP\n",
      "DEAPRMPEAAPPVAPAPAAPTPAAPAPAPSWPLSSSVPSQKTYQGSYGFRLGFLHSGTAK\n",
      "SVTCTYSPALNKMFCQLAKTCPVQLWVDSTPPPGTRVRAMAIYKQSQHMTEVVRRCPHHE\n",
      "RCSDSDGLAPPQHLIRVEGNLRVEYLDDRNTFRHSVVVPYEPPEVGSDCTTIHYNYMCNS\n",
      "SCMGGMNRRPILTIITLEDSSGNLLGRNSFEVRVCACPGRDRRTEEENLRKKGEPHHELP\n",
      "PGSTKRALPNNTSSSPQPKKKPLDGEYFTLQIRGRERFEMFRELNEALELKDAQAGKEPG\n",
      "GSRAHSSHLKSKKGQSTSRHKKLMFKTEGPDSD\n"
     ]
    }
   ],
   "source": [
    "./muscle -in p53.fas -fastaout p53.afa -phyiout p53.phy "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get ```phylip```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "engine": "bash",
    "eval": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-11-09 14:11:47--  http://evolution.gs.washington.edu/phylip/download/phylip-3.697.tar.gz\n",
      "Resolving evolution.gs.washington.edu (evolution.gs.washington.edu)... 128.208.27.135\n",
      "Connecting to evolution.gs.washington.edu (evolution.gs.washington.edu)|128.208.27.135|:80... connected.\n",
      "HTTP request sent, awaiting response... 302 Found\n",
      "Location: https://evolution.gs.washington.edu/phylip/download/phylip-3.697.tar.gz [following]\n",
      "--2021-11-09 14:11:47--  https://evolution.gs.washington.edu/phylip/download/phylip-3.697.tar.gz\n",
      "Connecting to evolution.gs.washington.edu (evolution.gs.washington.edu)|128.208.27.135|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 3958696 (3.8M) [application/x-gzip]\n",
      "Saving to: ‘phylip-3.697.tar.gz’\n",
      "\n",
      "100%[======================================>] 3,958,696   2.86MB/s   in 1.3s   \n",
      "\n",
      "2021-11-09 14:11:49 (2.86 MB/s) - ‘phylip-3.697.tar.gz’ saved [3958696/3958696]\n",
      "\n",
      "phylip-3.697/\n",
      "phylip-3.697/doc/\n",
      "phylip-3.697/doc/clique.html\n",
      "phylip-3.697/doc/consense.html\n",
      "phylip-3.697/doc/contchar.html\n",
      "phylip-3.697/doc/contml.html\n",
      "phylip-3.697/doc/contrast.html\n",
      "phylip-3.697/doc/discrete.html\n",
      "phylip-3.697/doc/distance.html\n",
      "phylip-3.697/doc/dnacomp.html\n",
      "phylip-3.697/doc/dnadist.html\n",
      "phylip-3.697/doc/dnainvar.html\n",
      "phylip-3.697/doc/dnaml.html\n",
      "phylip-3.697/doc/dnamlk.html\n",
      "phylip-3.697/doc/dnamove.html\n",
      "phylip-3.697/doc/dnapars.html\n",
      "phylip-3.697/doc/dnapenny.html\n",
      "phylip-3.697/doc/penny.html\n",
      "phylip-3.697/doc/dollop.html\n",
      "phylip-3.697/doc/dolmove.html\n",
      "phylip-3.697/doc/dolpenny.html\n",
      "phylip-3.697/doc/draw.html\n",
      "phylip-3.697/doc/drawgram.html\n",
      "phylip-3.697/doc/drawtree.html\n",
      "phylip-3.697/doc/factor.html\n",
      "phylip-3.697/doc/fitch.html\n",
      "phylip-3.697/doc/gendist.html\n",
      "phylip-3.697/doc/images/\n",
      "phylip-3.697/doc/images/DrawGramCat.png\n",
      "phylip-3.697/doc/images/DrawGramControls.png\n",
      "phylip-3.697/doc/images/DrawTreeCat.png\n",
      "phylip-3.697/doc/images/DrawTreeControls.png\n",
      "phylip-3.697/doc/images/InputTree.png\n",
      "phylip-3.697/doc/kitsch.html\n",
      "phylip-3.697/doc/main.html\n",
      "phylip-3.697/doc/mix.html\n",
      "phylip-3.697/doc/move.html\n",
      "phylip-3.697/doc/neighbor.html\n",
      "phylip-3.697/doc/pars.html\n",
      "phylip-3.697/doc/phylip.gif\n",
      "phylip-3.697/doc/proml.html\n",
      "phylip-3.697/doc/promlk.html\n",
      "phylip-3.697/doc/protdist.html\n",
      "phylip-3.697/doc/protpars.html\n",
      "phylip-3.697/doc/restdist.html\n",
      "phylip-3.697/doc/restml.html\n",
      "phylip-3.697/doc/retree.html\n",
      "phylip-3.697/doc/seqboot.html\n",
      "phylip-3.697/doc/sequence.html\n",
      "phylip-3.697/doc/treedist.html\n",
      "phylip-3.697/exe/\n",
      "phylip-3.697/phylip.html\n",
      "phylip-3.697/src/\n",
      "phylip-3.697/src/buildDmg.sh\n",
      "phylip-3.697/src/clique.c\n",
      "phylip-3.697/src/cons.c\n",
      "phylip-3.697/src/cons.h\n",
      "phylip-3.697/src/consense.c\n",
      "phylip-3.697/src/cont.c\n",
      "phylip-3.697/src/cont.h\n",
      "phylip-3.697/src/contml.c\n",
      "phylip-3.697/src/contrast.c\n",
      "phylip-3.697/src/COPYRIGHT\n",
      "phylip-3.697/src/disc.c\n",
      "phylip-3.697/src/disc.h\n",
      "phylip-3.697/src/discrete.c\n",
      "phylip-3.697/src/discrete.h\n",
      "phylip-3.697/src/dist.c\n",
      "phylip-3.697/src/dist.h\n",
      "phylip-3.697/src/dnacomp.c\n",
      "phylip-3.697/src/dnadist.c\n",
      "phylip-3.697/src/dnainvar.c\n",
      "phylip-3.697/src/dnaml.c\n",
      "phylip-3.697/src/dnamlk.c\n",
      "phylip-3.697/src/dnamove.c\n",
      "phylip-3.697/src/dnapars.c\n",
      "phylip-3.697/src/dnapenny.c\n",
      "phylip-3.697/src/dollo.c\n",
      "phylip-3.697/src/dollo.h\n",
      "phylip-3.697/src/dollop.c\n",
      "phylip-3.697/src/dolmove.c\n",
      "phylip-3.697/src/dolpenny.c\n",
      "phylip-3.697/src/draw.c\n",
      "phylip-3.697/src/draw.h\n",
      "phylip-3.697/src/draw2.c\n",
      "phylip-3.697/src/drawgram.c\n",
      "phylip-3.697/src/drawtree.c\n",
      "phylip-3.697/src/factor.c\n",
      "phylip-3.697/src/fitch.c\n",
      "phylip-3.697/src/font1\n",
      "phylip-3.697/src/font2\n",
      "phylip-3.697/src/font3\n",
      "phylip-3.697/src/font4\n",
      "phylip-3.697/src/font5\n",
      "phylip-3.697/src/font6\n",
      "phylip-3.697/src/gendist.c\n",
      "phylip-3.697/src/icons/\n",
      "phylip-3.697/src/icons/boot.ico\n",
      "phylip-3.697/src/icons/clique.ico\n",
      "phylip-3.697/src/icons/clique.rc\n",
      "phylip-3.697/src/icons/clique.rcb\n",
      "phylip-3.697/src/icons/consense.ico\n",
      "phylip-3.697/src/icons/consense.rc\n",
      "phylip-3.697/src/icons/consense.rcb\n",
      "phylip-3.697/src/icons/contml.ico\n",
      "phylip-3.697/src/icons/contml.rc\n",
      "phylip-3.697/src/icons/contml.rcb\n",
      "phylip-3.697/src/icons/contrast.ico\n",
      "phylip-3.697/src/icons/contrast.rc\n",
      "phylip-3.697/src/icons/contrast.rcb\n",
      "phylip-3.697/src/icons/disc.ico\n",
      "phylip-3.697/src/icons/dist.ico\n",
      "phylip-3.697/src/icons/dna.ico\n",
      "phylip-3.697/src/icons/dnacomp.rc\n",
      "phylip-3.697/src/icons/dnacomp.rcb\n",
      "phylip-3.697/src/icons/dnadist.rc\n",
      "phylip-3.697/src/icons/dnadist.rcb\n",
      "phylip-3.697/src/icons/dnainvar.rc\n",
      "phylip-3.697/src/icons/dnainvar.rcb\n",
      "phylip-3.697/src/icons/dnaml.rc\n",
      "phylip-3.697/src/icons/dnaml.rcb\n",
      "phylip-3.697/src/icons/dnamlk.rc\n",
      "phylip-3.697/src/icons/dnamlk.rcb\n",
      "phylip-3.697/src/icons/dnamove.rc\n",
      "phylip-3.697/src/icons/dnamove.rcb\n",
      "phylip-3.697/src/icons/dnapars.rc\n",
      "phylip-3.697/src/icons/dnapars.rcb\n",
      "phylip-3.697/src/icons/dnapenny.rc\n",
      "phylip-3.697/src/icons/dnapenny.rcb\n",
      "phylip-3.697/src/icons/dollo.ico\n",
      "phylip-3.697/src/icons/dollop.rc\n",
      "phylip-3.697/src/icons/dollop.rcb\n",
      "phylip-3.697/src/icons/dolmove.rc\n",
      "phylip-3.697/src/icons/dolmove.rcb\n",
      "phylip-3.697/src/icons/dolpenny.rc\n",
      "phylip-3.697/src/icons/dolpenny.rcb\n",
      "phylip-3.697/src/icons/drawgram.ico\n",
      "phylip-3.697/src/icons/drawgram.rc\n",
      "phylip-3.697/src/icons/drawgram.rcb\n",
      "phylip-3.697/src/icons/drawtree.ico\n",
      "phylip-3.697/src/icons/drawtree.rc\n",
      "phylip-3.697/src/icons/drawtree.rcb\n",
      "phylip-3.697/src/icons/factor.rc\n",
      "phylip-3.697/src/icons/factor.rcb\n",
      "phylip-3.697/src/icons/fitch.rc\n",
      "phylip-3.697/src/icons/fitch.rcb\n",
      "phylip-3.697/src/icons/gendist.ico\n",
      "phylip-3.697/src/icons/gendist.rc\n",
      "phylip-3.697/src/icons/gendist.rcb\n",
      "phylip-3.697/src/icons/kitsch.rc\n",
      "phylip-3.697/src/icons/kitsch.rcb\n",
      "phylip-3.697/src/icons/mix.rc\n",
      "phylip-3.697/src/icons/mix.rcb\n",
      "phylip-3.697/src/icons/move.rc\n",
      "phylip-3.697/src/icons/move.rcb\n",
      "phylip-3.697/src/icons/neighbor.rc\n",
      "phylip-3.697/src/icons/neighbor.rcb\n",
      "phylip-3.697/src/icons/pars.rc\n",
      "phylip-3.697/src/icons/pars.rcb\n",
      "phylip-3.697/src/icons/penny.rc\n",
      "phylip-3.697/src/icons/penny.rcb\n",
      "phylip-3.697/src/icons/proml.rc\n",
      "phylip-3.697/src/icons/proml.rcb\n",
      "phylip-3.697/src/icons/promlk.rc\n",
      "phylip-3.697/src/icons/promlk.rcb\n",
      "phylip-3.697/src/icons/protdist.rc\n",
      "phylip-3.697/src/icons/protdist.rcb\n",
      "phylip-3.697/src/icons/protein.ico\n",
      "phylip-3.697/src/icons/protpars.rc\n",
      "phylip-3.697/src/icons/protpars.rcb\n",
      "phylip-3.697/src/icons/restdist.rc\n",
      "phylip-3.697/src/icons/restdist.rcb\n",
      "phylip-3.697/src/icons/restml.rc\n",
      "phylip-3.697/src/icons/restml.rcb\n",
      "phylip-3.697/src/icons/restrict.ico\n",
      "phylip-3.697/src/icons/retree.ico\n",
      "phylip-3.697/src/icons/retree.rc\n",
      "phylip-3.697/src/icons/retree.rcb\n",
      "phylip-3.697/src/icons/seqboot.rc\n",
      "phylip-3.697/src/icons/seqboot.rcb\n",
      "phylip-3.697/src/icons/treedist.ico\n",
      "phylip-3.697/src/icons/treedist.rc\n",
      "phylip-3.697/src/icons/treedist.rcb\n",
      "phylip-3.697/src/infile2\n",
      "phylip-3.697/src/io.h\n",
      "phylip-3.697/src/javajars/\n",
      "phylip-3.697/src/javajars/DrawGram.jar\n",
      "phylip-3.697/src/javajars/DrawGramJava.bat\n",
      "phylip-3.697/src/javajars/DrawGramJava.exe\n",
      "phylip-3.697/src/javajars/DrawGramJava.unx\n",
      "phylip-3.697/src/javajars/DrawTree.jar\n",
      "phylip-3.697/src/javajars/DrawTreeJava.bat\n",
      "phylip-3.697/src/javajars/DrawTreeJava.exe\n",
      "phylip-3.697/src/javajars/DrawTreeJava.unx\n",
      "phylip-3.697/src/javajars/libdrawgram.so\n",
      "phylip-3.697/src/javasrc/\n",
      "phylip-3.697/src/javasrc/drawgram/\n",
      "phylip-3.697/src/javasrc/drawgram/DrawgramInterface.java\n",
      "phylip-3.697/src/javasrc/drawgram/DrawgramUserInterface.java\n",
      "phylip-3.697/src/javasrc/drawtree/\n",
      "phylip-3.697/src/javasrc/drawtree/DrawtreeInterface.java\n",
      "phylip-3.697/src/javasrc/drawtree/DrawtreeUserInterface.java\n",
      "phylip-3.697/src/javasrc/util/\n",
      "phylip-3.697/src/javasrc/util/DisplayProgress.java\n",
      "phylip-3.697/src/javasrc/util/DrawPreview.java\n",
      "phylip-3.697/src/javasrc/util/LabelData.java\n",
      "phylip-3.697/src/javasrc/util/PhylipFileDialog.java\n",
      "phylip-3.697/src/javasrc/util/PlotData.java\n",
      "phylip-3.697/src/javasrc/util/SectionData.java\n",
      "phylip-3.697/src/javasrc/util/TestFileNames.java\n",
      "phylip-3.697/src/kitsch.c\n",
      "phylip-3.697/src/linkmac\n",
      "phylip-3.697/src/mac/\n",
      "phylip-3.697/src/mac/boot.icns\n",
      "phylip-3.697/src/mac/clique.icns\n",
      "phylip-3.697/src/mac/command.in\n",
      "phylip-3.697/src/mac/consense.icns\n",
      "phylip-3.697/src/mac/contml.icns\n",
      "phylip-3.697/src/mac/contrast.icns\n",
      "phylip-3.697/src/mac/disc.icns\n",
      "phylip-3.697/src/mac/dist.icns\n",
      "phylip-3.697/src/mac/dna.icns\n",
      "phylip-3.697/src/mac/dnacomp.icns\n",
      "phylip-3.697/src/mac/dnadist.icns\n",
      "phylip-3.697/src/mac/dnainvar.icns\n",
      "phylip-3.697/src/mac/dnaml.icns\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "phylip-3.697/src/mac/dnamlk.icns\n",
      "phylip-3.697/src/mac/dnamove.icns\n",
      "phylip-3.697/src/mac/dnapars.icns\n",
      "phylip-3.697/src/mac/dnapenny.icns\n",
      "phylip-3.697/src/mac/dollo.icns\n",
      "phylip-3.697/src/mac/dollop.icns\n",
      "phylip-3.697/src/mac/dolmove.icns\n",
      "phylip-3.697/src/mac/dolpenny.icns\n",
      "phylip-3.697/src/mac/drawgram.icns\n",
      "phylip-3.697/src/mac/drawtree.icns\n",
      "phylip-3.697/src/mac/factor.icns\n",
      "phylip-3.697/src/mac/fitch.icns\n",
      "phylip-3.697/src/mac/gendist.icns\n",
      "phylip-3.697/src/mac/Info.plist.in\n",
      "phylip-3.697/src/mac/kitsch.icns\n",
      "phylip-3.697/src/mac/mix.icns\n",
      "phylip-3.697/src/mac/move.icns\n",
      "phylip-3.697/src/mac/neighbor.icns\n",
      "phylip-3.697/src/mac/pars.icns\n",
      "phylip-3.697/src/mac/penny.icns\n",
      "phylip-3.697/src/mac/proml.icns\n",
      "phylip-3.697/src/mac/promlk.icns\n",
      "phylip-3.697/src/mac/protdist.icns\n",
      "phylip-3.697/src/mac/protein.icns\n",
      "phylip-3.697/src/mac/protpars.icns\n",
      "phylip-3.697/src/mac/restdist.icns\n",
      "phylip-3.697/src/mac/restml.icns\n",
      "phylip-3.697/src/mac/restrict.icns\n",
      "phylip-3.697/src/mac/retree.icns\n",
      "phylip-3.697/src/mac/seqboot.icns\n",
      "phylip-3.697/src/mac/treedist.icns\n",
      "phylip-3.697/src/Makefile.cyg\n",
      "phylip-3.697/src/Makefile.osx\n",
      "phylip-3.697/src/Makefile.unx\n",
      "phylip-3.697/src/mix.c\n",
      "phylip-3.697/src/mlclock.c\n",
      "phylip-3.697/src/mlclock.h\n",
      "phylip-3.697/src/move.c\n",
      "phylip-3.697/src/moves.c\n",
      "phylip-3.697/src/moves.h\n",
      "phylip-3.697/src/neighbor.c\n",
      "phylip-3.697/src/pars.c\n",
      "phylip-3.697/src/penny.c\n",
      "phylip-3.697/src/phylip.c\n",
      "phylip-3.697/src/phylip.h\n",
      "phylip-3.697/src/phylip.html\n",
      "phylip-3.697/src/printree.c\n",
      "phylip-3.697/src/printree.h\n",
      "phylip-3.697/src/proml.c\n",
      "phylip-3.697/src/promlk.c\n",
      "phylip-3.697/src/protdist.c\n",
      "phylip-3.697/src/protpars.c\n",
      "phylip-3.697/src/restdist.c\n",
      "phylip-3.697/src/restml.c\n",
      "phylip-3.697/src/retree.c\n",
      "phylip-3.697/src/seq.c\n",
      "phylip-3.697/src/seq.h\n",
      "phylip-3.697/src/seqboot.c\n",
      "phylip-3.697/src/treedist.c\n",
      "phylip-3.697/src/wagner.c\n",
      "phylip-3.697/src/wagner.h\n",
      "Building PHYLIP version 3.696\n",
      "gcc    -c -o clique.o clique.c\n",
      "gcc    -c -o disc.o disc.c\n",
      "gcc    -c -o phylip.o phylip.c\n",
      "gcc  clique.o disc.o phylip.o -lm -o clique\n",
      "gcc    -c -o consense.o consense.c\n",
      "gcc    -c -o cons.o cons.c\n",
      "gcc  consense.o phylip.o cons.o -lm -o consense\n",
      "gcc    -c -o contml.o contml.c\n",
      "gcc    -c -o cont.o cont.c\n",
      "gcc  contml.o cont.o phylip.o -lm -o contml\n",
      "gcc    -c -o contrast.o contrast.c\n",
      "gcc  contrast.o cont.o phylip.o -lm -o contrast\n",
      "gcc    -c -o dnacomp.o dnacomp.c\n",
      "gcc    -c -o seq.o seq.c\n",
      "gcc  dnacomp.o seq.o phylip.o -lm -o dnacomp\n",
      "gcc    -c -o dnadist.o dnadist.c\n",
      "gcc  dnadist.o seq.o phylip.o -lm -o dnadist\n",
      "gcc    -c -o dnainvar.o dnainvar.c\n",
      "gcc  dnainvar.o seq.o phylip.o -lm -o dnainvar\n",
      "gcc    -c -o dnaml.o dnaml.c\n",
      "gcc  dnaml.o seq.o phylip.o -lm -o dnaml\n",
      "gcc    -c -o dnamlk.o dnamlk.c\n",
      "gcc    -c -o mlclock.o mlclock.c\n",
      "gcc    -c -o printree.o printree.c\n",
      "gcc  dnamlk.o seq.o phylip.o mlclock.o printree.o -lm -o dnamlk\n",
      "gcc    -c -o dnamove.o dnamove.c\n",
      "gcc    -c -o moves.o moves.c\n",
      "gcc  dnamove.o seq.o moves.o phylip.o -lm -o dnamove\n",
      "gcc    -c -o dnapars.o dnapars.c\n",
      "gcc  dnapars.o seq.o phylip.o -lm -o dnapars\n",
      "gcc    -c -o dnapenny.o dnapenny.c\n",
      "gcc  dnapenny.o seq.o phylip.o -lm -o dnapenny\n",
      "gcc    -c -o dolmove.o dolmove.c\n",
      "gcc    -c -o dollo.o dollo.c\n",
      "gcc  dolmove.o disc.o moves.o dollo.o phylip.o -lm -o dolmove\n",
      "gcc    -c -o dollop.o dollop.c\n",
      "gcc  dollop.o disc.o dollo.o phylip.o -lm -o dollop\n",
      "gcc    -c -o dolpenny.o dolpenny.c\n",
      "gcc  dolpenny.o disc.o dollo.o phylip.o -lm -o dolpenny\n",
      "gcc    -c -o factor.o factor.c\n",
      "gcc  factor.o phylip.o -lm -o factor\n",
      "gcc    -c -o fitch.o fitch.c\n",
      "gcc    -c -o dist.o dist.c\n",
      "gcc  fitch.o dist.o phylip.o -lm -o fitch\n",
      "gcc    -c -o gendist.o gendist.c\n",
      "gcc  gendist.o phylip.o -lm -o gendist\n",
      "gcc    -c -o kitsch.o kitsch.c\n",
      "gcc  kitsch.o dist.o phylip.o -lm -o kitsch\n",
      "gcc    -c -o mix.o mix.c\n",
      "gcc    -c -o wagner.o wagner.c\n",
      "gcc  mix.o disc.o wagner.o phylip.o -lm -o mix\n",
      "gcc    -c -o move.o move.c\n",
      "gcc  move.o disc.o moves.o wagner.o phylip.o -lm -o move\n",
      "gcc    -c -o neighbor.o neighbor.c\n",
      "gcc  neighbor.o dist.o phylip.o -lm -o neighbor\n",
      "gcc    -c -o pars.o pars.c\n",
      "gcc    -c -o discrete.o discrete.c\n",
      "gcc  pars.o discrete.o phylip.o -lm -o pars\n",
      "gcc    -c -o penny.o penny.c\n",
      "gcc  penny.o disc.o wagner.o  phylip.o -lm -o penny\n",
      "gcc    -c -o proml.o proml.c\n",
      "gcc  proml.o seq.o phylip.o -lm -o proml\n",
      "gcc    -c -o promlk.o promlk.c\n",
      "gcc  promlk.o seq.o phylip.o mlclock.o printree.o -lm -o promlk\n",
      "gcc    -c -o protdist.o protdist.c\n",
      "gcc  protdist.o seq.o phylip.o -lm -o protdist\n",
      "gcc    -c -o protpars.o protpars.c\n",
      "gcc  protpars.o seq.o phylip.o -lm -o protpars\n",
      "gcc    -c -o restdist.o restdist.c\n",
      "gcc  restdist.o seq.o phylip.o -lm -o restdist\n",
      "gcc    -c -o restml.o restml.c\n",
      "gcc  restml.o seq.o phylip.o -lm -o restml\n",
      "gcc    -c -o retree.o retree.c\n",
      "gcc  retree.o moves.o phylip.o -lm -o retree\n",
      "gcc    -c -o seqboot.o seqboot.c\n",
      "gcc  seqboot.o seq.o phylip.o -lm -o seqboot\n",
      "gcc    -c -o treedist.o treedist.c\n",
      "gcc  treedist.o cons.o phylip.o -lm -o treedist\n",
      "gcc  -c drawgram.c\n",
      "drawgram.c: In function ‘drawgram’:\n",
      "drawgram.c:2111:9: warning: implicit declaration of function ‘makebox’; did you mean ‘makedev’? [-Wimplicit-function-declaration]\n",
      "         makebox(fontname,&xoffset,&yoffset,&scale,spp);\n",
      "         ^~~~~~~\n",
      "         makedev\n",
      "gcc  -c draw.c\n",
      "gcc  -c draw2.c\n",
      "gcc  draw.o draw2.o drawgram.o phylip.o -lm -o drawgram\n",
      "gcc   -shared -fPIC -c drawtree.c\n",
      "drawtree.c: In function ‘drawtree’:\n",
      "drawtree.c:3017:9: warning: implicit declaration of function ‘makebox’; did you mean ‘makedev’? [-Wimplicit-function-declaration]\n",
      "         makebox(fontname,&xoffset,&yoffset,&scale,spp);\n",
      "         ^~~~~~~\n",
      "         makedev\n",
      "gcc  draw.o draw2.o drawtree.o phylip.o -lm -o drawtree\n",
      "gcc  -o libdrawgram.so -shared -fPIC drawgram.c draw.c draw2.c phylip.c \n",
      "drawgram.c: In function ‘drawgram’:\n",
      "drawgram.c:2111:9: warning: implicit declaration of function ‘makebox’; did you mean ‘makedev’? [-Wimplicit-function-declaration]\n",
      "         makebox(fontname,&xoffset,&yoffset,&scale,spp);\n",
      "         ^~~~~~~\n",
      "         makedev\n",
      "gcc  -o libdrawtree.so  -shared -fPIC drawtree.c draw.c draw2.c phylip.c \n",
      "drawtree.c: In function ‘drawtree’:\n",
      "drawtree.c:3017:9: warning: implicit declaration of function ‘makebox’; did you mean ‘makedev’? [-Wimplicit-function-declaration]\n",
      "         makebox(fontname,&xoffset,&yoffset,&scale,spp);\n",
      "         ^~~~~~~\n",
      "         makedev\n",
      "Finished compiling.\n",
      "\n",
      "Installing PHYLIP v3.6 binaries in ../exe\n",
      "Installing dynamic libraries in ../exe\n",
      "Installing jar files in ../exe\n",
      "Installing font files in ../exe\n",
      "Finished installation.\n",
      "\n",
      "Removing object files to save space\n",
      "Finished removing object files.  Now will remove\n",
      "executable files from the current directory, but not from the\n",
      "executables directory.  (If some are not here, the makefile\n",
      "will terminate with an error message but this is not a problem)\n",
      "\n",
      "Removing executables from this directory\n",
      "Finished cleanup.\n",
      "\n",
      "Done.\n",
      "\n",
      "/data/user/home/jziad/info_302_class.git/SeqSearch\n",
      "/data/user/home/jziad/info_302_class.git/SeqSearch/phylip-3.697/exe/protdist\n"
     ]
    }
   ],
   "source": [
    "wget http://evolution.gs.washington.edu/phylip/download/phylip-3.697.tar.gz\n",
    "tar -xvzf phylip-3.697.tar.gz\n",
    "cd phylip-3.697/src\n",
    "#pwd\n",
    "make -f Makefile.unx install\n",
    "cd ../exe\n",
    "export PATH=$PATH:`pwd`\n",
    "cd ../..\n",
    "pwd\n",
    "which protdist"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Running ```protdist```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "engine": "bash",
    "eval": false,
    "lines_to_next_cell": 0
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[2J\u001b[H\n",
      "\u001b[2J\u001b[H\n",
      "Protein distance algorithm, version 3.697\n",
      "\n",
      "Settings for this run:\n",
      "  P  Use JTT, PMB, PAM, Kimura, categories model?  Jones-Taylor-Thornton matrix\n",
      "  G  Gamma distribution of rates among positions?  No\n",
      "  C           One category of substitution rates?  Yes\n",
      "  W                    Use weights for positions?  No\n",
      "  M                   Analyze multiple data sets?  No\n",
      "  I                  Input sequences interleaved?  Yes\n",
      "  0                 Terminal type (IBM PC, ANSI)?  ANSI\n",
      "  1            Print out the data at start of run  No\n",
      "  2          Print indications of progress of run  Yes\n",
      "\n",
      "Are these settings correct? (type Y or the letter for one to change)\n",
      "\u001b[2J\u001b[H\n",
      "Protein distance algorithm, version 3.697\n",
      "\n",
      "Settings for this run:\n",
      "  P  Use JTT, PMB, PAM, Kimura, categories model?  Henikoff/Tillier PMB matrix\n",
      "  G  Gamma distribution of rates among positions?  No\n",
      "  C           One category of substitution rates?  Yes\n",
      "  W                    Use weights for positions?  No\n",
      "  M                   Analyze multiple data sets?  No\n",
      "  I                  Input sequences interleaved?  Yes\n",
      "  0                 Terminal type (IBM PC, ANSI)?  ANSI\n",
      "  1            Print out the data at start of run  No\n",
      "  2          Print indications of progress of run  Yes\n",
      "\n",
      "Are these settings correct? (type Y or the letter for one to change)\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "Done.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cp p53.phy infile\n",
    "/data/user/home/jziad/info_302_class.git/SeqSearch/phylip-3.697/exe/protdist <<EOF\n",
    "p53.phy\n",
    "y\n",
    "EOF\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This will create the distance matrix in a file ```outfile```. We need to rename this file so that it can be read my `neighbor`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "#rm infile\n",
    "mv outfile infile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "\n",
    "## Running ```neighbor```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "engine": "bash",
    "eval": false,
    "lines_to_next_cell": 0
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[2J\u001b[H\n",
      "\u001b[2J\u001b[H\n",
      "Neighbor-Joining/UPGMA method version 3.697\n",
      "\n",
      "Settings for this run:\n",
      "  N       Neighbor-joining or UPGMA tree?  Neighbor-joining\n",
      "  O                        Outgroup root?  No, use as outgroup species  1\n",
      "  L         Lower-triangular data matrix?  No\n",
      "  R         Upper-triangular data matrix?  No\n",
      "  S                        Subreplicates?  No\n",
      "  J     Randomize input order of species?  No. Use input order\n",
      "  M           Analyze multiple data sets?  No\n",
      "  0   Terminal type (IBM PC, ANSI, none)?  ANSI\n",
      "  1    Print out the data at start of run  No\n",
      "  2  Print indications of progress of run  Yes\n",
      "  3                        Print out tree  Yes\n",
      "  4       Write out trees onto tree file?  Yes\n",
      "\n",
      "\n",
      "  Y to accept these or type the letter for one to change\n",
      "\n",
      "Cycle   5: species 7 (   0.29151) joins species 8 (   0.28297)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.02111) joins species 6 (   0.02267)\n",
      "Cycle   2: species 1 (   0.13681) joins node 7 (   0.22066)\n",
      "Cycle   1: species 3 (   0.09694) joins node 4 (   0.06129)\n",
      "last cycle:\n",
      " node 1  (   0.02842) joins species 2  (   0.11497) joins node 3  (   0.00966)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Done.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "/data/user/home/jziad/info_302_class.git/SeqSearch/phylip-3.697/exe/neighbor <<EOF\n",
    "y\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The tree will be produced in a file called ```outtree```:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "engine": "bash",
    "eval": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "((NP_0012587:0.29151,NP_0010019:0.28297):0.22066,(NP_776626.:0.11497,\n",
      "(NP_0010032:0.09694,((NP_000537.:0.00001,XP_0011720:0.00000):0.02111,\n",
      "NP_0010406:0.02267):0.06129):0.00966):0.02842,NP_112251.:0.13681);\n"
     ]
    }
   ],
   "source": [
    "cat outtree"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is a newick format tree.\n",
    "\n",
    "## Viewing the tree\n",
    "To view the tree we can use ```FigTree```. You can download it from here:\n",
    "\n",
    "http://tree.bio.ed.ac.uk/software/figtree/\n",
    "\n",
    "Download and run the software as\n",
    "\n",
    "`java -jar figtree.jar`\n",
    "\n",
    "![Tree generated using ```neighbor```](figures/outtree.pdf)\n",
    "\n",
    "We can also use R package ape to view the file. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Bootstrapping\n",
    "\n",
    "To get the statistical measurement of tree, we can use \"bootstrapping\". We have to start with the raw ```phylip``` file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "engine": "bash",
    "eval": false,
    "lines_to_next_cell": 0
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[2J\u001b[H\u001b[2J\u001b[H\n",
      "Bootstrapping algorithm, version 3.697\n",
      "\n",
      "Settings for this run:\n",
      "  D      Sequence, Morph, Rest., Gene Freqs?  Molecular sequences\n",
      "  J  Bootstrap, Jackknife, Permute, Rewrite?  Bootstrap\n",
      "  %    Regular or altered sampling fraction?  regular\n",
      "  B      Block size for block-bootstrapping?  1 (regular bootstrap)\n",
      "  R                     How many replicates?  100\n",
      "  W              Read weights of characters?  No\n",
      "  C                Read categories of sites?  No\n",
      "  S     Write out data sets or just weights?  Data sets\n",
      "  I             Input sequences interleaved?  Yes\n",
      "  0      Terminal type (IBM PC, ANSI, none)?  ANSI\n",
      "  1       Print out the data at start of run  No\n",
      "  2     Print indications of progress of run  Yes\n",
      "\n",
      "  Y to accept these or type the letter for one to change\n",
      "\n",
      "Random number seed (must be odd)?\n",
      "\n",
      "completed replicate number   10\n",
      "completed replicate number   20\n",
      "completed replicate number   30\n",
      "completed replicate number   40\n",
      "completed replicate number   50\n",
      "completed replicate number   60\n",
      "completed replicate number   70\n",
      "completed replicate number   80\n",
      "completed replicate number   90\n",
      "completed replicate number  100\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "Done.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cd phylip-3.697/exe\n",
    "export PATH=$PATH:`pwd`\n",
    "cd ../..\n",
    "cp p53.phy infile\n",
    "rm infile\n",
    "cp outtree njtree.tre\n",
    "cp p53.phy infile\n",
    "seqboot<<EOF\n",
    "y\n",
    "3\n",
    "EOF\n",
    "rm infile\n",
    "mv outfile infile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now follow the exact step described earlier. Don't forget to change the option **\"Multple data sets\"**. The treefile will now have 100 trees (if you have used 100 bootstrap datasets). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[2J\u001b[H\n",
      "\u001b[2J\u001b[H\n",
      "Protein distance algorithm, version 3.697\n",
      "\n",
      "Settings for this run:\n",
      "  P  Use JTT, PMB, PAM, Kimura, categories model?  Jones-Taylor-Thornton matrix\n",
      "  G  Gamma distribution of rates among positions?  No\n",
      "  C           One category of substitution rates?  Yes\n",
      "  W                    Use weights for positions?  No\n",
      "  M                   Analyze multiple data sets?  No\n",
      "  I                  Input sequences interleaved?  Yes\n",
      "  0                 Terminal type (IBM PC, ANSI)?  ANSI\n",
      "  1            Print out the data at start of run  No\n",
      "  2          Print indications of progress of run  Yes\n",
      "\n",
      "Are these settings correct? (type Y or the letter for one to change)\n",
      "Multiple data sets or multiple weights? (type D or W)\n",
      "How many data sets?\n",
      "\u001b[2J\u001b[H\n",
      "Protein distance algorithm, version 3.697\n",
      "\n",
      "Settings for this run:\n",
      "  P  Use JTT, PMB, PAM, Kimura, categories model?  Jones-Taylor-Thornton matrix\n",
      "  G  Gamma distribution of rates among positions?  No\n",
      "  C           One category of substitution rates?  Yes\n",
      "  W                    Use weights for positions?  No\n",
      "  M                   Analyze multiple data sets?  Yes, 100 data sets\n",
      "  I                  Input sequences interleaved?  Yes\n",
      "  0                 Terminal type (IBM PC, ANSI)?  ANSI\n",
      "  1            Print out the data at start of run  No\n",
      "  2          Print indications of progress of run  Yes\n",
      "\n",
      "Are these settings correct? (type Y or the letter for one to change)\n",
      "\n",
      "\n",
      "Data set # 1:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 2:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 3:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 4:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 5:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 6:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 7:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 8:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 9:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 10:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 11:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 12:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 13:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 14:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 15:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 16:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 17:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 18:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 19:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 20:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 21:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 22:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 23:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 24:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 25:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 26:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 27:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 28:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 29:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 30:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 31:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 32:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 33:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 34:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 35:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 36:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 37:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 38:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 39:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 40:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 41:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 42:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 43:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 44:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 45:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 46:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 47:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 48:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 49:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 50:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 51:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 52:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 53:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 54:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 55:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 56:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 57:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 58:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 59:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 60:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 61:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 62:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 63:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 64:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 65:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 66:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 67:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 68:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 69:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 70:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 71:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 72:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 73:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 74:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 75:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 76:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 77:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 78:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 79:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 80:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 81:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 82:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 83:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 84:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 85:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 86:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 87:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 88:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 89:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 90:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 91:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 92:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 93:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 94:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 95:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 96:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 97:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 98:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 99:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "\n",
      "\n",
      "Data set # 100:\n",
      "\n",
      "Computing distances:\n",
      "  NP_112251.   \n",
      "  NP_776626.   .\n",
      "  NP_0010032   ..\n",
      "  NP_000537.   ...\n",
      "  XP_0011720   ....\n",
      "  NP_0010406   .....\n",
      "  NP_0012587   ......\n",
      "  NP_0010019   .......\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "Done.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "/data/user/home/jziad/info_302_class.git/SeqSearch/phylip-3.697/exe/protdist <<EOF\n",
    "M\n",
    "D\n",
    "100\n",
    "y\n",
    "EOF\n",
    "rm infile\n",
    "mv outfile infile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run neighbor again"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[2J\u001b[H\n",
      "\u001b[2J\u001b[H\n",
      "Neighbor-Joining/UPGMA method version 3.697\n",
      "\n",
      "Settings for this run:\n",
      "  N       Neighbor-joining or UPGMA tree?  Neighbor-joining\n",
      "  O                        Outgroup root?  No, use as outgroup species  1\n",
      "  L         Lower-triangular data matrix?  No\n",
      "  R         Upper-triangular data matrix?  No\n",
      "  S                        Subreplicates?  No\n",
      "  J     Randomize input order of species?  No. Use input order\n",
      "  M           Analyze multiple data sets?  No\n",
      "  0   Terminal type (IBM PC, ANSI, none)?  ANSI\n",
      "  1    Print out the data at start of run  No\n",
      "  2  Print indications of progress of run  Yes\n",
      "  3                        Print out tree  Yes\n",
      "  4       Write out trees onto tree file?  Yes\n",
      "\n",
      "\n",
      "  Y to accept these or type the letter for one to change\n",
      "How many data sets?\n",
      "\n",
      "Random number seed (must be odd)?\n",
      "\u001b[2J\u001b[H\n",
      "Neighbor-Joining/UPGMA method version 3.697\n",
      "\n",
      "Settings for this run:\n",
      "  N       Neighbor-joining or UPGMA tree?  Neighbor-joining\n",
      "  O                        Outgroup root?  No, use as outgroup species  1\n",
      "  L         Lower-triangular data matrix?  No\n",
      "  R         Upper-triangular data matrix?  No\n",
      "  S                        Subreplicates?  No\n",
      "  J     Randomize input order of species?  Yes (random number seed =       3)\n",
      "  M           Analyze multiple data sets?  Yes, 100 sets\n",
      "  0   Terminal type (IBM PC, ANSI, none)?  ANSI\n",
      "  1    Print out the data at start of run  No\n",
      "  2  Print indications of progress of run  Yes\n",
      "  3                        Print out tree  Yes\n",
      "  4       Write out trees onto tree file?  Yes\n",
      "\n",
      "\n",
      "  Y to accept these or type the letter for one to change\n",
      "Data set # 1:\n",
      "\n",
      "Cycle   5: species 7 (   0.37289) joins species 8 (   0.35313)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.03084) joins species 6 (   0.03192)\n",
      "Cycle   2: node 7 (   0.24114) joins species 1 (   0.14613)\n",
      "Cycle   1: node 7 (   0.03464) joins species 3 (   0.10874)\n",
      "last cycle:\n",
      " species 2  (   0.11536) joins node 4  (   0.04129) joins node 7  (   0.00515)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 2:\n",
      "\n",
      "Cycle   5: species 8 (   0.38921) joins species 7 (   0.41391)\n",
      "Cycle   4: species 1 (   0.13543) joins node 8 (   0.15067)\n",
      "Cycle   3: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   2: node 4 (   0.02085) joins species 6 (   0.02443)\n",
      "Cycle   1: species 3 (   0.10559) joins species 2 (   0.10441)\n",
      "last cycle:\n",
      " node 1  (   0.05894) joins node 3  (   0.00671) joins node 4  (   0.05040)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 3:\n",
      "\n",
      "Cycle   5: species 7 (   0.28664) joins species 8 (   0.29779)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.01740) joins species 6 (   0.01907)\n",
      "Cycle   2: node 4 (   0.05505) joins species 1 (   0.16324)\n",
      "Cycle   1: node 7 (   0.28629) joins node 4 (   0.02659)\n",
      "last cycle:\n",
      " species 2  (   0.10866) joins species 3  (   0.07868) joins node 7  (   0.01247)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 4:\n",
      "\n",
      "Cycle   5: species 8 (   0.37097) joins species 7 (   0.35853)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.02608) joins species 6 (   0.03490)\n",
      "Cycle   2: node 4 (   0.05198) joins species 1 (   0.19083)\n",
      "Cycle   1: species 2 (   0.09702) joins node 8 (   0.29007)\n",
      "last cycle:\n",
      " node 2  (   0.00721) joins species 3  (   0.10393) joins node 4  (   0.01556)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 5:\n",
      "\n",
      "Cycle   5: species 8 (   0.36059) joins species 7 (   0.31802)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: species 6 (   0.01094) joins node 5 (   0.01289)\n",
      "Cycle   2: node 8 (   0.21787) joins species 1 (   0.10787)\n",
      "Cycle   1: species 3 (   0.09143) joins node 6 (   0.06064)\n",
      "last cycle:\n",
      " species 2  (   0.10734) joins node 3  (   0.01302) joins node 8  (   0.02324)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 6:\n",
      "\n",
      "Cycle   5: species 8 (   0.29984) joins species 7 (   0.31722)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.01786) joins species 6 (   0.01309)\n",
      "Cycle   2: species 1 (   0.13211) joins node 4 (   0.06418)\n",
      "Cycle   1: node 1 (   0.01025) joins species 2 (   0.12700)\n",
      "last cycle:\n",
      " node 1  (   0.00597) joins species 3  (   0.10477) joins node 8  (   0.37837)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 7:\n",
      "\n",
      "Cycle   5: species 8 (   0.40004) joins species 7 (   0.42662)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02175) joins species 6 (   0.02167)\n",
      "Cycle   2: node 8 (   0.27949) joins species 2 (   0.12764)\n",
      "Cycle   1: node 5 (   0.05758) joins species 3 (   0.13382)\n",
      "last cycle:\n",
      " species 1  (   0.17265) joins node 5  (   0.00886) joins node 8  (   0.00904)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 8:\n",
      "\n",
      "Cycle   5: species 8 (   0.26007) joins species 7 (   0.37818)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: species 6 (   0.01854) joins node 5 (   0.02731)\n",
      "Cycle   2: species 1 (   0.14699) joins node 8 (   0.27875)\n",
      "Cycle   1: node 1 (   0.02338) joins species 2 (   0.10145)\n",
      "last cycle:\n",
      " node 1  (   0.01637) joins species 3  (   0.13043) joins node 6  (   0.02829)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 9:\n",
      "\n",
      "Cycle   5: species 7 (   0.33279) joins species 8 (   0.29321)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02474) joins species 6 (   0.02116)\n",
      "Cycle   2: node 7 (   0.31731) joins species 1 (   0.16815)\n",
      "Cycle   1: node 5 (   0.07150) joins species 3 (   0.12127)\n",
      "last cycle:\n",
      " species 2  (   0.13705) joins node 5  (   0.00089) joins node 7  (   0.02302)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 10:\n",
      "\n",
      "Cycle   5: species 7 (   0.40569) joins species 8 (   0.34028)\n",
      "Cycle   4: node 7 (   0.27372) joins species 1 (   0.14718)\n",
      "Cycle   3: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   2: node 5 (   0.02842) joins species 6 (   0.02443)\n",
      "Cycle   1: node 7 (   0.04127) joins species 2 (   0.10249)\n",
      "last cycle:\n",
      " species 3  (   0.09641) joins node 5  (   0.05473) joins node 7  (   0.01743)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 11:\n",
      "\n",
      "Cycle   5: species 8 (   0.28756) joins species 7 (   0.28819)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.01244) joins species 6 (   0.02547)\n",
      "Cycle   2: node 8 (   0.27711) joins species 1 (   0.15137)\n",
      "Cycle   1: node 8 (   0.02403) joins species 2 (   0.10424)\n",
      "last cycle:\n",
      " species 3  (   0.09782) joins node 5  (   0.06438) joins node 8  (   0.01758)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 12:\n",
      "\n",
      "Cycle   5: species 7 (   0.34408) joins species 8 (   0.35078)\n",
      "Cycle   4: node 7 (   0.26673) joins species 1 (   0.13120)\n",
      "Cycle   3: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   2: node 4 (   0.02389) joins species 6 (   0.02684)\n",
      "Cycle   1: species 2 (   0.10108) joins node 7 (   0.04931)\n",
      "last cycle:\n",
      " node 2  (   0.01065) joins species 3  (   0.10752) joins node 4  (   0.03944)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 13:\n",
      "\n",
      "Cycle   5: species 7 (   0.34100) joins species 8 (   0.36224)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02532) joins species 6 (   0.03854)\n",
      "Cycle   2: node 7 (   0.24650) joins species 1 (   0.13869)\n",
      "Cycle   1: species 2 (   0.16441) joins species 3 (   0.09929)\n",
      "last cycle:\n",
      " node 2  (   0.01193) joins node 5  (   0.04891) joins node 7  (   0.03899)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 14:\n",
      "\n",
      "Cycle   5: species 8 (   0.34385) joins species 7 (   0.32100)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.00892) joins species 6 (   0.01720)\n",
      "Cycle   2: node 4 (   0.06440) joins species 3 (   0.09024)\n",
      "Cycle   1: node 8 (   0.25040) joins species 1 (   0.15160)\n",
      "last cycle:\n",
      " species 2  (   0.11555) joins node 4  (   0.02031) joins node 8  (   0.00960)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 15:\n",
      "\n",
      "Cycle   5: species 7 (   0.39945) joins species 8 (   0.27915)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.03291) joins species 6 (   0.02797)\n",
      "Cycle   2: species 3 (   0.11838) joins species 2 (   0.11575)\n",
      "Cycle   1: species 1 (   0.15847) joins node 5 (   0.07594)\n",
      "last cycle:\n",
      " node 1  (   0.02344) joins node 3  (   0.02783) joins node 7  (   0.23138)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 16:\n",
      "\n",
      "Cycle   5: species 8 (   0.26519) joins species 7 (   0.34769)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.01233) joins species 6 (   0.02077)\n",
      "Cycle   2: species 1 (   0.16835) joins node 8 (   0.23220)\n",
      "Cycle   1: species 3 (   0.09006) joins species 2 (   0.08897)\n",
      "last cycle:\n",
      " node 1  (   0.02367) joins node 3  (   0.00550) joins node 4  (   0.06747)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 17:\n",
      "\n",
      "Cycle   5: species 7 (   0.37812) joins species 8 (   0.31575)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: species 6 (   0.01590) joins node 5 (   0.01014)\n",
      "Cycle   2: species 1 (   0.13328) joins node 7 (   0.29607)\n",
      "Cycle   1: node 1 (   0.01548) joins node 6 (   0.07276)\n",
      "last cycle:\n",
      " node 1  (   0.01522) joins species 2  (   0.11553) joins species 3  (   0.09688)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 18:\n",
      "\n",
      "Cycle   5: species 8 (   0.32212) joins species 7 (   0.37215)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: species 6 (   0.01761) joins node 5 (   0.01635)\n",
      "Cycle   2: node 8 (   0.21071) joins species 1 (   0.11783)\n",
      "Cycle   1: node 6 (   0.05878) joins node 8 (   0.02099)\n",
      "last cycle:\n",
      " species 2  (   0.11104) joins species 3  (   0.09807) joins node 6  (   0.00646)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 19:\n",
      "\n",
      "Cycle   5: species 8 (   0.32552) joins species 7 (   0.32644)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02907) joins species 6 (   0.02469)\n",
      "Cycle   2: species 3 (   0.06646) joins species 2 (   0.13045)\n",
      "Cycle   1: node 3 (   0.00714) joins node 5 (   0.06116)\n",
      "last cycle:\n",
      " species 1  (   0.17162) joins node 3  (   0.00182) joins node 8  (   0.33326)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 20:\n",
      "\n",
      "Cycle   5: species 7 (   0.35017) joins species 8 (   0.35223)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.01836) joins species 6 (   0.01293)\n",
      "Cycle   2: species 1 (   0.15249) joins node 7 (   0.24363)\n",
      "Cycle   1: node 4 (   0.07099) joins node 1 (   0.03302)\n",
      "last cycle:\n",
      " species 2  (   0.10787) joins species 3  (   0.11843) joins node 4  (   0.00967)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 21:\n",
      "\n",
      "Cycle   5: species 8 (   0.27367) joins species 7 (   0.37871)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.01890) joins species 6 (   0.01613)\n",
      "Cycle   2: node 8 (   0.25377) joins species 2 (   0.10979)\n",
      "Cycle   1: species 3 (   0.09728) joins node 4 (   0.07241)\n",
      "last cycle:\n",
      " species 1  (   0.13361) joins node 3  (   0.01874) joins node 8  (   0.02696)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 22:\n",
      "\n",
      "Cycle   5: species 8 (   0.36855) joins species 7 (   0.37831)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: species 6 (   0.02671) joins node 5 (   0.02324)\n",
      "Cycle   2: node 8 (   0.28506) joins species 1 (   0.16152)\n",
      "Cycle   1: node 8 (   0.01699) joins species 2 (   0.11924)\n",
      "last cycle:\n",
      " species 3  (   0.11221) joins node 6  (   0.06745) joins node 8  (   0.01002)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 23:\n",
      "\n",
      "Cycle   5: species 8 (   0.29139) joins species 7 (   0.32454)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.01674) joins species 6 (   0.03498)\n",
      "Cycle   2: node 8 (   0.26511) joins species 3 (   0.07939)\n",
      "Cycle   1: node 8 (   0.02050) joins species 2 (   0.12700)\n",
      "last cycle:\n",
      " species 1  (   0.13426) joins node 4  (   0.06270) joins node 8  (   0.01146)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 24:\n",
      "\n",
      "Cycle   5: species 8 (   0.33301) joins species 7 (   0.38718)\n",
      "Cycle   4: species 1 (   0.16115) joins node 8 (   0.32962)\n",
      "Cycle   3: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   2: node 5 (   0.01629) joins species 6 (   0.01804)\n",
      "Cycle   1: species 3 (   0.09834) joins node 1 (   0.05174)\n",
      "last cycle:\n",
      " species 2  (   0.08305) joins node 3  (   0.00588) joins node 5  (   0.05757)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 25:\n",
      "\n",
      "Cycle   5: species 8 (   0.33219) joins species 7 (   0.32004)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: species 6 (   0.01673) joins node 5 (   0.01562)\n",
      "Cycle   2: species 2 (   0.10356) joins node 8 (   0.29138)\n",
      "Cycle   1: node 6 (   0.06095) joins species 1 (   0.17147)\n",
      "last cycle:\n",
      " node 2  (   0.01258) joins species 3  (   0.09850) joins node 6  (   0.00911)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 26:\n",
      "\n",
      "Cycle   5: species 8 (   0.38265) joins species 7 (   0.38058)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.03998) joins species 6 (   0.02546)\n",
      "Cycle   2: node 8 (   0.35294) joins species 1 (   0.18620)\n",
      "Cycle   1: node 5 (   0.04913) joins species 3 (   0.09755)\n",
      "last cycle:\n",
      " species 2  (   0.10084) joins node 5  (   0.01112) joins node 8  (   0.01853)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 27:\n",
      "\n",
      "Cycle   5: species 8 (   0.31045) joins species 7 (   0.32788)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.01373) joins species 6 (   0.01549)\n",
      "Cycle   2: node 4 (   0.08220) joins species 3 (   0.10390)\n",
      "Cycle   1: node 8 (   0.21068) joins species 1 (   0.16860)\n",
      "last cycle:\n",
      " species 2  (   0.11562) joins node 4  (   0.01940) joins node 8  (   0.01775)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 28:\n",
      "\n",
      "Cycle   5: species 8 (   0.30235) joins species 7 (   0.30793)\n",
      "Cycle   4: node 8 (   0.22976) joins species 1 (   0.11880)\n",
      "Cycle   3: node 8 (   0.04297) joins species 2 (   0.09591)\n",
      "Cycle   2: species 3 (   0.06573) joins node 8 (   0.02318)\n",
      "Cycle   1: species 6 (   0.03215) joins node 3 (   0.04962)\n",
      "last cycle:\n",
      " species 4  (   0.00001) joins species 5  (   0.00000) joins node 6  (   0.02244)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 29:\n",
      "\n",
      "Cycle   5: species 8 (   0.35964) joins species 7 (   0.36876)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.03120) joins species 6 (   0.02891)\n",
      "Cycle   2: node 8 (   0.24967) joins species 3 (   0.10900)\n",
      "Cycle   1: node 8 (   0.03161) joins species 2 (   0.13524)\n",
      "last cycle:\n",
      " species 1  (   0.14431) joins node 4  (   0.04860) joins node 8  (   0.00984)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 30:\n",
      "\n",
      "Cycle   5: species 7 (   0.33162) joins species 8 (   0.34583)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: species 6 (   0.01797) joins node 4 (   0.01547)\n",
      "Cycle   2: node 6 (   0.05703) joins species 3 (   0.10292)\n",
      "Cycle   1: node 7 (   0.27619) joins species 1 (   0.14671)\n",
      "last cycle:\n",
      " species 2  (   0.13841) joins node 6  (   0.01110) joins node 7  (   0.00596)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 31:\n",
      "\n",
      "Cycle   5: species 7 (   0.35256) joins species 8 (   0.34212)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: species 6 (   0.03432) joins node 4 (   0.02617)\n",
      "Cycle   2: species 2 (   0.12783) joins species 3 (   0.10946)\n",
      "Cycle   1: node 7 (   0.21456) joins species 1 (   0.14168)\n",
      "last cycle:\n",
      " node 2  (   0.02609) joins node 6  (   0.07222) joins node 7  (   0.01619)\n",
      "\n",
      "Output written on file \"outfile\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 32:\n",
      "\n",
      "Cycle   5: species 8 (   0.31991) joins species 7 (   0.30318)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02951) joins species 6 (   0.02226)\n",
      "Cycle   2: species 1 (   0.13981) joins node 8 (   0.28701)\n",
      "Cycle   1: node 1 (   0.02683) joins species 2 (   0.14463)\n",
      "last cycle:\n",
      " node 1  (   0.01351) joins species 3  (   0.10483) joins node 5  (   0.05143)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 33:\n",
      "\n",
      "Cycle   5: species 7 (   0.31433) joins species 8 (   0.34066)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.02401) joins species 6 (   0.02971)\n",
      "Cycle   2: species 1 (   0.15836) joins node 7 (   0.21331)\n",
      "Cycle   1: node 1 (   0.03104) joins species 2 (   0.14353)\n",
      "last cycle:\n",
      " node 1  (   0.01496) joins species 3  (   0.11278) joins node 4  (   0.07056)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 34:\n",
      "\n",
      "Cycle   5: species 8 (   0.26617) joins species 7 (   0.33918)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.03934) joins species 6 (   0.03337)\n",
      "Cycle   2: node 8 (   0.31801) joins species 1 (   0.12530)\n",
      "Cycle   1: node 8 (   0.03388) joins species 3 (   0.10004)\n",
      "last cycle:\n",
      " species 2  (   0.09795) joins node 4  (   0.04166) joins node 8  (   0.00621)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 35:\n",
      "\n",
      "Cycle   5: species 8 (   0.34290) joins species 7 (   0.28851)\n",
      "Cycle   4: node 8 (   0.31614) joins species 1 (   0.09467)\n",
      "Cycle   3: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   2: node 4 (   0.02698) joins species 6 (   0.03745)\n",
      "Cycle   1: node 8 (   0.05587) joins node 4 (   0.06102)\n",
      "last cycle:\n",
      " species 2  (   0.11708) joins species 3  (   0.07418) joins node 8  (   0.01748)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 36:\n",
      "\n",
      "Cycle   5: species 8 (   0.35321) joins species 7 (   0.33136)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02271) joins species 6 (   0.02336)\n",
      "Cycle   2: species 1 (   0.14696) joins node 8 (   0.29820)\n",
      "Cycle   1: node 1 (   0.03243) joins species 2 (   0.14486)\n",
      "last cycle:\n",
      " node 1  (   0.01049) joins species 3  (   0.12400) joins node 5  (   0.05452)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 37:\n",
      "\n",
      "Cycle   5: species 7 (   0.33463) joins species 8 (   0.27090)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.01723) joins species 6 (   0.02282)\n",
      "Cycle   2: species 1 (   0.13799) joins node 7 (   0.32197)\n",
      "Cycle   1: node 4 (   0.07820) joins node 1 (   0.03002)\n",
      "last cycle:\n",
      " species 2  (   0.08030) joins species 3  (   0.08466) joins node 4  (   0.02555)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 38:\n",
      "\n",
      "Cycle   5: species 8 (   0.35750) joins species 7 (   0.34376)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.01788) joins species 6 (   0.01473)\n",
      "Cycle   2: node 4 (   0.04402) joins species 3 (   0.08864)\n",
      "Cycle   1: node 8 (   0.29203) joins species 2 (   0.10144)\n",
      "last cycle:\n",
      " species 1  (   0.11922) joins node 4  (   0.02633) joins node 8  (   0.00980)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 39:\n",
      "\n",
      "Cycle   5: species 7 (   0.29164) joins species 8 (   0.33884)\n",
      "Cycle   4: node 7 (   0.28778) joins species 1 (   0.13601)\n",
      "Cycle   3: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   2: node 4 (   0.01226) joins species 6 (   0.02672)\n",
      "Cycle   1: node 7 (   0.03786) joins node 4 (   0.05862)\n",
      "last cycle:\n",
      " species 2  (   0.08504) joins species 3  (   0.11012) joins node 7  (   0.00365)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 40:\n",
      "\n",
      "Cycle   5: species 7 (   0.33844) joins species 8 (   0.25618)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.03199) joins species 6 (   0.03019)\n",
      "Cycle   2: node 7 (   0.27348) joins species 1 (   0.08877)\n",
      "Cycle   1: node 5 (   0.05564) joins node 7 (   0.01722)\n",
      "last cycle:\n",
      " species 2  (   0.13999) joins species 3  (   0.05169) joins node 5  (   0.01664)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 41:\n",
      "\n",
      "Cycle   5: species 7 (   0.33229) joins species 8 (   0.33785)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02274) joins species 6 (   0.02738)\n",
      "Cycle   2: node 7 (   0.25526) joins species 1 (   0.12845)\n",
      "Cycle   1: node 5 (   0.04991) joins node 7 (   0.02697)\n",
      "last cycle:\n",
      " species 2  (   0.09308) joins species 3  (   0.12577) joins node 5  (   0.00302)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 42:\n",
      "\n",
      "Cycle   5: species 8 (   0.29297) joins species 7 (   0.33109)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.02251) joins species 6 (   0.02142)\n",
      "Cycle   2: node 8 (   0.28518) joins species 1 (   0.13577)\n",
      "Cycle   1: species 3 (   0.07504) joins node 4 (   0.05508)\n",
      "last cycle:\n",
      " species 2  (   0.10610) joins node 3  (   0.01059) joins node 8  (   0.01574)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 43:\n",
      "\n",
      "Cycle   5: species 7 (   0.33586) joins species 8 (   0.30667)\n",
      "Cycle   4: node 7 (   0.19977) joins species 1 (   0.14702)\n",
      "Cycle   3: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   2: node 5 (   0.02405) joins species 6 (   0.01890)\n",
      "Cycle   1: species 2 (   0.12191) joins node 7 (   0.05606)\n",
      "last cycle:\n",
      " node 2  (   0.02106) joins species 3  (   0.11415) joins node 5  (   0.04555)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 44:\n",
      "\n",
      "Cycle   5: species 8 (   0.36808) joins species 7 (   0.37244)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.01974) joins species 6 (   0.02256)\n",
      "Cycle   2: species 1 (   0.16284) joins node 8 (   0.25199)\n",
      "Cycle   1: node 1 (   0.03764) joins species 2 (   0.12036)\n",
      "last cycle:\n",
      " node 1  (   0.02455) joins species 3  (   0.09363) joins node 4  (   0.07522)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 45:\n",
      "\n",
      "Cycle   5: species 7 (   0.36928) joins species 8 (   0.29494)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02252) joins species 6 (   0.01642)\n",
      "Cycle   2: node 7 (   0.27551) joins species 1 (   0.12558)\n",
      "Cycle   1: species 2 (   0.11016) joins node 7 (   0.02682)\n",
      "last cycle:\n",
      " node 2  (   0.02409) joins species 3  (   0.10216) joins node 5  (   0.07182)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 46:\n",
      "\n",
      "Cycle   5: species 7 (   0.39056) joins species 8 (   0.32351)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.02219) joins species 6 (   0.02372)\n",
      "Cycle   2: species 3 (   0.08145) joins node 4 (   0.06116)\n",
      "Cycle   1: species 1 (   0.16612) joins node 7 (   0.23591)\n",
      "last cycle:\n",
      " node 1  (   0.03091) joins species 2  (   0.13155) joins node 3  (   0.03218)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 47:\n",
      "\n",
      "Cycle   5: species 8 (   0.44165) joins species 7 (   0.36465)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: species 6 (   0.02395) joins node 4 (   0.01987)\n",
      "Cycle   2: node 6 (   0.06679) joins species 1 (   0.15426)\n",
      "Cycle   1: node 6 (   0.03070) joins species 3 (   0.08277)\n",
      "last cycle:\n",
      " species 2  (   0.12926) joins node 6  (   0.00914) joins node 8  (   0.26991)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 48:\n",
      "\n",
      "Cycle   5: species 7 (   0.35774) joins species 8 (   0.29381)\n",
      "Cycle   4: species 1 (   0.13030) joins node 7 (   0.24386)\n",
      "Cycle   3: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   2: node 5 (   0.01920) joins species 6 (   0.01233)\n",
      "Cycle   1: node 5 (   0.09043) joins species 3 (   0.09943)\n",
      "last cycle:\n",
      " node 1  (   0.05998) joins species 2  (   0.08955) joins node 5  (   0.00530)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Data set # 49:\n",
      "\n",
      "Cycle   5: species 8 (   0.28759) joins species 7 (   0.32173)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.01650) joins species 6 (   0.01938)\n",
      "Cycle   2: node 5 (   0.05123) joins species 3 (   0.07943)\n",
      "Cycle   1: species 2 (   0.11449) joins node 5 (   0.02735)\n",
      "last cycle:\n",
      " species 1  (   0.16776) joins node 2  (   0.01076) joins node 8  (   0.23146)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 50:\n",
      "\n",
      "Cycle   5: species 8 (   0.34178) joins species 7 (   0.36890)\n",
      "Cycle   4: node 8 (   0.16513) joins species 1 (   0.18345)\n",
      "Cycle   3: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   2: species 6 (   0.02204) joins node 4 (   0.01151)\n",
      "Cycle   1: species 2 (   0.12897) joins node 8 (   0.02712)\n",
      "last cycle:\n",
      " node 2  (   0.01134) joins species 3  (   0.09553) joins node 6  (   0.05117)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 51:\n",
      "\n",
      "Cycle   5: species 7 (   0.34660) joins species 8 (   0.30294)\n",
      "Cycle   4: node 7 (   0.22577) joins species 1 (   0.15446)\n",
      "Cycle   3: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   2: species 6 (   0.01410) joins node 4 (   0.01193)\n",
      "Cycle   1: species 2 (   0.13954) joins node 7 (   0.05176)\n",
      "last cycle:\n",
      " node 2  (   0.00884) joins species 3  (   0.10307) joins node 6  (   0.06465)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 52:\n",
      "\n",
      "Cycle   5: species 7 (   0.26988) joins species 8 (   0.29583)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02825) joins species 6 (   0.02023)\n",
      "Cycle   2: node 5 (   0.07272) joins species 3 (   0.06094)\n",
      "Cycle   1: node 5 (   0.00996) joins species 2 (   0.12964)\n",
      "last cycle:\n",
      " species 1  (   0.16371) joins node 5  (   0.00601) joins node 7  (   0.24344)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 53:\n",
      "\n",
      "Cycle   5: species 7 (   0.38432) joins species 8 (   0.30200)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.02084) joins species 6 (   0.03172)\n",
      "Cycle   2: species 1 (   0.13146) joins node 7 (   0.24452)\n",
      "Cycle   1: node 1 (   0.02090) joins species 2 (   0.09469)\n",
      "last cycle:\n",
      " node 1  (   0.00538) joins species 3  (   0.09057) joins node 4  (   0.07255)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 54:\n",
      "\n",
      "Cycle   5: species 8 (   0.39256) joins species 7 (   0.35132)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02217) joins species 6 (   0.03380)\n",
      "Cycle   2: node 8 (   0.25093) joins species 1 (   0.18253)\n",
      "Cycle   1: node 5 (   0.06070) joins species 3 (   0.10277)\n",
      "last cycle:\n",
      " species 2  (   0.10460) joins node 5  (   0.00164) joins node 8  (   0.03083)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 55:\n",
      "\n",
      "Cycle   5: species 7 (   0.37341) joins species 8 (   0.33643)\n",
      "Cycle   4: species 1 (   0.12464) joins node 7 (   0.28190)\n",
      "Cycle   3: node 1 (   0.04030) joins species 2 (   0.08886)\n",
      "Cycle   2: node 1 (   0.01862) joins species 3 (   0.09781)\n",
      "Cycle   1: node 1 (   0.06060) joins species 6 (   0.02161)\n",
      "last cycle:\n",
      " node 1  (   0.01393) joins species 4  (   0.00001) joins species 5  (   0.00001)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 56:\n",
      "\n",
      "Cycle   5: species 7 (   0.29818) joins species 8 (   0.31474)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.02072) joins species 6 (   0.04476)\n",
      "Cycle   2: species 1 (   0.16288) joins node 7 (   0.32556)\n",
      "Cycle   1: node 4 (   0.04587) joins species 3 (   0.10351)\n",
      "last cycle:\n",
      " node 1  (   0.02178) joins species 2  (   0.12808) joins node 4  (   0.01290)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 57:\n",
      "\n",
      "Cycle   5: species 7 (   0.42789) joins species 8 (   0.34714)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.01469) joins species 6 (   0.01418)\n",
      "Cycle   2: node 5 (   0.06939) joins species 3 (   0.11769)\n",
      "Cycle   1: node 5 (   0.01734) joins species 2 (   0.10489)\n",
      "last cycle:\n",
      " species 1  (   0.18264) joins node 5  (   0.00775) joins node 7  (   0.22887)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 58:\n",
      "\n",
      "Cycle   5: species 7 (   0.32504) joins species 8 (   0.32324)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.02130) joins species 6 (   0.01811)\n",
      "Cycle   2: node 7 (   0.21144) joins species 1 (   0.14528)\n",
      "Cycle   1: species 3 (   0.10056) joins node 4 (   0.04326)\n",
      "last cycle:\n",
      " species 2  (   0.11507) joins node 3  (   0.00697) joins node 7  (   0.01789)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 59:\n",
      "\n",
      "Cycle   5: species 8 (   0.33993) joins species 7 (   0.32971)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.01704) joins species 6 (   0.01189)\n",
      "Cycle   2: node 8 (   0.23474) joins species 1 (   0.16383)\n",
      "Cycle   1: node 5 (   0.05108) joins node 8 (   0.02243)\n",
      "last cycle:\n",
      " species 2  (   0.09916) joins species 3  (   0.12922) joins node 5  (   0.00906)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 60:\n",
      "\n",
      "Cycle   5: species 8 (   0.32398) joins species 7 (   0.35666)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02029) joins species 6 (   0.01088)\n",
      "Cycle   2: species 1 (   0.15229) joins node 8 (   0.22513)\n",
      "Cycle   1: node 1 (   0.02444) joins species 2 (   0.09211)\n",
      "last cycle:\n",
      " node 1  (   0.01146) joins species 3  (   0.13763) joins node 5  (   0.07631)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 61:\n",
      "\n",
      "Cycle   5: species 8 (   0.29027) joins species 7 (   0.31724)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: species 6 (   0.01938) joins node 4 (   0.02004)\n",
      "Cycle   2: species 3 (   0.10024) joins node 6 (   0.04911)\n",
      "Cycle   1: node 8 (   0.29895) joins species 2 (   0.11591)\n",
      "last cycle:\n",
      " species 1  (   0.17087) joins node 3  (   0.02130) joins node 8  (   0.01002)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 62:\n",
      "\n",
      "Cycle   5: species 7 (   0.38898) joins species 8 (   0.27387)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.03045) joins species 6 (   0.02367)\n",
      "Cycle   2: node 7 (   0.23508) joins species 1 (   0.15304)\n",
      "Cycle   1: node 7 (   0.02124) joins species 2 (   0.11687)\n",
      "last cycle:\n",
      " species 3  (   0.08846) joins node 4  (   0.06607) joins node 7  (   0.00938)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 63:\n",
      "\n",
      "Cycle   5: species 7 (   0.40858) joins species 8 (   0.35851)\n",
      "Cycle   4: node 7 (   0.26773) joins species 1 (   0.12936)\n",
      "Cycle   3: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   2: node 5 (   0.01217) joins species 6 (   0.02158)\n",
      "Cycle   1: species 3 (   0.10387) joins node 7 (   0.05269)\n",
      "last cycle:\n",
      " species 2  (   0.10996) joins node 3  (   0.00547) joins node 5  (   0.08015)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 64:\n",
      "\n",
      "Cycle   5: species 7 (   0.35230) joins species 8 (   0.33081)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.01594) joins species 6 (   0.01202)\n",
      "Cycle   2: node 7 (   0.29959) joins species 1 (   0.14035)\n",
      "Cycle   1: node 7 (   0.03065) joins species 2 (   0.10953)\n",
      "last cycle:\n",
      " species 3  (   0.11418) joins node 5  (   0.05488) joins node 7  (   0.00874)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 65:\n",
      "\n",
      "Cycle   5: species 7 (   0.38546) joins species 8 (   0.36044)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.02493) joins species 6 (   0.02182)\n",
      "Cycle   2: species 1 (   0.11846) joins node 7 (   0.27745)\n",
      "Cycle   1: species 2 (   0.13737) joins node 1 (   0.02439)\n",
      "last cycle:\n",
      " node 2  (   0.00578) joins species 3  (   0.09490) joins node 4  (   0.06379)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 66:\n",
      "\n",
      "Cycle   5: species 8 (   0.33832) joins species 7 (   0.28496)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02311) joins species 6 (   0.01272)\n",
      "Cycle   2: node 8 (   0.23806) joins species 1 (   0.13238)\n",
      "Cycle   1: species 3 (   0.08517) joins node 5 (   0.05057)\n",
      "last cycle:\n",
      " species 2  (   0.12074) joins node 3  (   0.00446) joins node 8  (   0.03597)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 67:\n",
      "\n",
      "Cycle   5: species 7 (   0.41585) joins species 8 (   0.38035)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.01655) joins species 6 (   0.02027)\n",
      "Cycle   2: species 1 (   0.17222) joins node 7 (   0.25935)\n",
      "Cycle   1: node 1 (   0.03120) joins node 4 (   0.07438)\n",
      "last cycle:\n",
      " node 1  (   0.00747) joins species 2  (   0.12791) joins species 3  (   0.11620)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 68:\n",
      "\n",
      "Cycle   5: species 7 (   0.27756) joins species 8 (   0.28062)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.02754) joins species 6 (   0.01445)\n",
      "Cycle   2: node 4 (   0.07731) joins species 3 (   0.11538)\n",
      "Cycle   1: node 7 (   0.30592) joins species 1 (   0.17716)\n",
      "last cycle:\n",
      " species 2  (   0.11842) joins node 4  (   0.01345) joins node 7  (   0.00785)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 69:\n",
      "\n",
      "Cycle   5: species 7 (   0.30438) joins species 8 (   0.38803)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.02382) joins species 6 (   0.01711)\n",
      "Cycle   2: node 7 (   0.27509) joins species 1 (   0.10400)\n",
      "Cycle   1: node 4 (   0.04485) joins species 3 (   0.06483)\n",
      "last cycle:\n",
      " species 2  (   0.11882) joins node 4  (   0.01636) joins node 7  (   0.03564)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 70:\n",
      "\n",
      "Cycle   5: species 7 (   0.36186) joins species 8 (   0.23806)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.03138) joins species 6 (   0.03688)\n",
      "Cycle   2: species 1 (   0.11559) joins node 7 (   0.28816)\n",
      "Cycle   1: node 1 (   0.04939) joins node 5 (   0.05448)\n",
      "last cycle:\n",
      " node 1  (   0.00976) joins species 2  (   0.10659) joins species 3  (   0.07007)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 71:\n",
      "\n",
      "Cycle   5: species 8 (   0.28089) joins species 7 (   0.34006)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02531) joins species 6 (   0.02975)\n",
      "Cycle   2: species 3 (   0.08892) joins species 2 (   0.10602)\n",
      "Cycle   1: node 8 (   0.28002) joins species 1 (   0.15618)\n",
      "last cycle:\n",
      " node 3  (   0.00775) joins node 5  (   0.04541) joins node 8  (   0.00731)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 72:\n",
      "\n",
      "Cycle   5: species 7 (   0.29297) joins species 8 (   0.29910)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02435) joins species 6 (   0.01873)\n",
      "Cycle   2: node 7 (   0.25121) joins species 1 (   0.11593)\n",
      "Cycle   1: node 5 (   0.06204) joins species 2 (   0.09810)\n",
      "last cycle:\n",
      " species 3  (   0.09300) joins node 5  (   0.00839) joins node 7  (   0.04095)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 73:\n",
      "\n",
      "Cycle   5: species 7 (   0.40352) joins species 8 (   0.35550)\n",
      "Cycle   4: node 7 (   0.28034) joins species 1 (   0.16327)\n",
      "Cycle   3: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   2: node 4 (   0.02423) joins species 6 (   0.02912)\n",
      "Cycle   1: node 7 (   0.03949) joins species 2 (   0.10973)\n",
      "last cycle:\n",
      " species 3  (   0.08696) joins node 4  (   0.03989) joins node 7  (   0.01857)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 74:\n",
      "\n",
      "Cycle   5: species 7 (   0.34870) joins species 8 (   0.35264)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02797) joins species 6 (   0.03196)\n",
      "Cycle   2: node 7 (   0.26083) joins species 1 (   0.16860)\n",
      "Cycle   1: node 5 (   0.06828) joins node 7 (   0.02268)\n",
      "last cycle:\n",
      " species 2  (   0.13793) joins species 3  (   0.11565) joins node 5  (   0.01515)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 75:\n",
      "\n",
      "Cycle   5: species 7 (   0.35479) joins species 8 (   0.36409)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: species 6 (   0.01399) joins node 4 (   0.01476)\n",
      "Cycle   2: node 7 (   0.35125) joins species 2 (   0.11821)\n",
      "Cycle   1: species 1 (   0.16835) joins node 7 (   0.02684)\n",
      "last cycle:\n",
      " node 1  (   0.00734) joins species 3  (   0.09869) joins node 6  (   0.05363)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 76:\n",
      "\n",
      "Cycle   5: species 7 (   0.33380) joins species 8 (   0.26386)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: species 6 (   0.02034) joins node 5 (   0.02156)\n",
      "Cycle   2: node 7 (   0.23653) joins species 1 (   0.11437)\n",
      "Cycle   1: node 6 (   0.06470) joins node 7 (   0.02305)\n",
      "last cycle:\n",
      " species 2  (   0.11031) joins species 3  (   0.07159) joins node 6  (   0.00969)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 77:\n",
      "\n",
      "Cycle   5: species 8 (   0.34946) joins species 7 (   0.28850)\n",
      "Cycle   4: node 8 (   0.28747) joins species 1 (   0.10836)\n",
      "Cycle   3: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   2: node 4 (   0.02270) joins species 6 (   0.02480)\n",
      "Cycle   1: node 4 (   0.03972) joins node 8 (   0.04612)\n",
      "last cycle:\n",
      " species 2  (   0.10865) joins species 3  (   0.09987) joins node 4  (   0.00920)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 78:\n",
      "\n",
      "Cycle   5: species 7 (   0.32171) joins species 8 (   0.33477)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02369) joins species 6 (   0.01850)\n",
      "Cycle   2: node 5 (   0.06853) joins species 1 (   0.14795)\n",
      "Cycle   1: species 3 (   0.11158) joins species 2 (   0.10473)\n",
      "last cycle:\n",
      " node 3  (   0.00341) joins node 5  (   0.01234) joins node 7  (   0.28364)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 79:\n",
      "\n",
      "Cycle   5: species 7 (   0.32721) joins species 8 (   0.39653)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: species 6 (   0.01892) joins node 5 (   0.03006)\n",
      "Cycle   2: species 2 (   0.11538) joins species 3 (   0.09252)\n",
      "Cycle   1: species 1 (   0.14233) joins node 7 (   0.27715)\n",
      "last cycle:\n",
      " node 1  (   0.01547) joins node 2  (   0.01980) joins node 6  (   0.06394)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 80:\n",
      "\n",
      "Cycle   5: species 8 (   0.33694) joins species 7 (   0.27856)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.02661) joins species 6 (   0.02399)\n",
      "Cycle   2: node 8 (   0.22598) joins species 2 (   0.08957)\n",
      "Cycle   1: node 8 (   0.02593) joins species 1 (   0.14290)\n",
      "last cycle:\n",
      " species 3  (   0.10262) joins node 4  (   0.05111) joins node 8  (   0.00707)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 81:\n",
      "\n",
      "Cycle   5: species 7 (   0.34677) joins species 8 (   0.30225)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: species 6 (   0.02399) joins node 4 (   0.01282)\n",
      "Cycle   2: species 1 (   0.14350) joins node 7 (   0.20994)\n",
      "Cycle   1: node 6 (   0.05644) joins species 3 (   0.08988)\n",
      "last cycle:\n",
      " node 1  (   0.00947) joins species 2  (   0.10707) joins node 6  (   0.00596)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 82:\n",
      "\n",
      "Cycle   5: species 7 (   0.36888) joins species 8 (   0.31325)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.03239) joins species 6 (   0.02723)\n",
      "Cycle   2: species 3 (   0.10722) joins node 4 (   0.06221)\n",
      "Cycle   1: node 7 (   0.25956) joins species 1 (   0.17192)\n",
      "last cycle:\n",
      " species 2  (   0.11005) joins node 3  (   0.01298) joins node 7  (   0.00431)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 83:\n",
      "\n",
      "Cycle   5: species 8 (   0.27597) joins species 7 (   0.38299)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cycle   3: node 5 (   0.02275) joins species 6 (   0.01928)\n",
      "Cycle   2: node 8 (   0.28112) joins species 1 (   0.12784)\n",
      "Cycle   1: species 3 (   0.07877) joins node 5 (   0.04358)\n",
      "last cycle:\n",
      " species 2  (   0.11847) joins node 3  (   0.00672) joins node 8  (   0.02956)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 84:\n",
      "\n",
      "Cycle   5: species 8 (   0.34901) joins species 7 (   0.41557)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02154) joins species 6 (   0.02973)\n",
      "Cycle   2: node 8 (   0.23631) joins species 1 (   0.15315)\n",
      "Cycle   1: node 8 (   0.01473) joins species 3 (   0.10253)\n",
      "last cycle:\n",
      " species 2  (   0.11972) joins node 5  (   0.04680) joins node 8  (   0.01013)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 85:\n",
      "\n",
      "Cycle   5: species 8 (   0.32407) joins species 7 (   0.37339)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02158) joins species 6 (   0.01927)\n",
      "Cycle   2: node 8 (   0.25788) joins species 3 (   0.14222)\n",
      "Cycle   1: node 5 (   0.05895) joins species 1 (   0.17355)\n",
      "last cycle:\n",
      " species 2  (   0.08732) joins node 5  (   0.01692) joins node 8  (   0.02771)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 86:\n",
      "\n",
      "Cycle   5: species 7 (   0.33372) joins species 8 (   0.34102)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02616) joins species 6 (   0.02699)\n",
      "Cycle   2: node 7 (   0.23187) joins species 1 (   0.15885)\n",
      "Cycle   1: node 7 (   0.03072) joins species 2 (   0.11117)\n",
      "last cycle:\n",
      " species 3  (   0.07942) joins node 5  (   0.08020) joins node 7  (   0.01138)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 87:\n",
      "\n",
      "Cycle   5: species 8 (   0.32171) joins species 7 (   0.29144)\n",
      "Cycle   4: species 1 (   0.11075) joins node 8 (   0.25679)\n",
      "Cycle   3: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   2: node 5 (   0.02932) joins species 6 (   0.02623)\n",
      "Cycle   1: node 1 (   0.05235) joins node 5 (   0.05893)\n",
      "last cycle:\n",
      " node 1  (   0.00465) joins species 2  (   0.11561) joins species 3  (   0.09581)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 88:\n",
      "\n",
      "Cycle   5: species 8 (   0.32887) joins species 7 (   0.31271)\n",
      "Cycle   4: node 8 (   0.23920) joins species 1 (   0.14976)\n",
      "Cycle   3: node 8 (   0.02323) joins species 2 (   0.08213)\n",
      "Cycle   2: node 8 (   0.02594) joins species 3 (   0.11596)\n",
      "Cycle   1: node 8 (   0.03018) joins species 6 (   0.02714)\n",
      "last cycle:\n",
      " species 4  (   0.00001) joins species 5  (   0.00000) joins node 8  (   0.01286)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 89:\n",
      "\n",
      "Cycle   5: species 8 (   0.26548) joins species 7 (   0.38930)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02723) joins species 6 (   0.02324)\n",
      "Cycle   2: node 8 (   0.26872) joins species 1 (   0.14790)\n",
      "Cycle   1: node 5 (   0.04134) joins species 3 (   0.13752)\n",
      "last cycle:\n",
      " species 2  (   0.12004) joins node 5  (   0.00927) joins node 8  (   0.01216)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 90:\n",
      "\n",
      "Cycle   5: species 7 (   0.35311) joins species 8 (   0.32850)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.02355) joins species 6 (   0.03109)\n",
      "Cycle   2: node 7 (   0.31693) joins species 1 (   0.15166)\n",
      "Cycle   1: node 7 (   0.02570) joins species 2 (   0.12427)\n",
      "last cycle:\n",
      " species 3  (   0.11629) joins node 4  (   0.05978) joins node 7  (   0.00141)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 91:\n",
      "\n",
      "Cycle   5: species 7 (   0.31884) joins species 8 (   0.30774)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: species 6 (   0.02009) joins node 4 (   0.01999)\n",
      "Cycle   2: species 1 (   0.12090) joins node 7 (   0.24929)\n",
      "Cycle   1: node 1 (   0.03717) joins node 6 (   0.07149)\n",
      "last cycle:\n",
      " node 1  (   0.00672) joins species 2  (   0.14294) joins species 3  (   0.07467)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 92:\n",
      "\n",
      "Cycle   5: species 8 (   0.28611) joins species 7 (   0.41604)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02671) joins species 6 (   0.02424)\n",
      "Cycle   2: node 8 (   0.31152) joins species 1 (   0.15302)\n",
      "Cycle   1: species 2 (   0.13576) joins node 8 (   0.02705)\n",
      "last cycle:\n",
      " node 2  (   0.00743) joins species 3  (   0.09326) joins node 5  (   0.06504)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 93:\n",
      "\n",
      "Cycle   5: species 8 (   0.27162) joins species 7 (   0.41255)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.01782) joins species 6 (   0.01632)\n",
      "Cycle   2: node 8 (   0.24178) joins species 2 (   0.11008)\n",
      "Cycle   1: node 8 (   0.02499) joins node 5 (   0.06478)\n",
      "last cycle:\n",
      " species 1  (   0.18055) joins species 3  (   0.10211) joins node 8  (   0.00264)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 94:\n",
      "\n",
      "Cycle   5: species 7 (   0.34750) joins species 8 (   0.36048)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.02171) joins species 6 (   0.02670)\n",
      "Cycle   2: node 7 (   0.31217) joins species 1 (   0.15578)\n",
      "Cycle   1: species 2 (   0.16449) joins node 7 (   0.02446)\n",
      "last cycle:\n",
      " node 2  (   0.01783) joins species 3  (   0.12925) joins node 4  (   0.08720)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 95:\n",
      "\n",
      "Cycle   5: species 8 (   0.27946) joins species 7 (   0.37703)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: node 5 (   0.02649) joins species 6 (   0.03376)\n",
      "Cycle   2: species 1 (   0.14177) joins node 8 (   0.24007)\n",
      "Cycle   1: species 3 (   0.12967) joins node 5 (   0.04652)\n",
      "last cycle:\n",
      " node 1  (   0.03365) joins species 2  (   0.10141) joins node 3  (   0.02414)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 96:\n",
      "\n",
      "Cycle   5: species 8 (   0.31386) joins species 7 (   0.36070)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.02511) joins species 6 (   0.02446)\n",
      "Cycle   2: node 8 (   0.27040) joins species 1 (   0.16090)\n",
      "Cycle   1: node 4 (   0.06122) joins node 8 (   0.00864)\n",
      "last cycle:\n",
      " species 2  (   0.10508) joins species 3  (   0.08958) joins node 4  (   0.00342)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 97:\n",
      "\n",
      "Cycle   5: species 7 (   0.36062) joins species 8 (   0.36934)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: species 6 (   0.02743) joins node 4 (   0.02413)\n",
      "Cycle   2: node 7 (   0.26290) joins species 1 (   0.16804)\n",
      "Cycle   1: species 2 (   0.13655) joins node 7 (   0.03079)\n",
      "last cycle:\n",
      " node 2  (   0.01970) joins species 3  (   0.12566) joins node 6  (   0.06116)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 98:\n",
      "\n",
      "Cycle   5: species 8 (   0.38106) joins species 7 (   0.42454)\n",
      "Cycle   4: species 4 (   0.00001) joins species 5 (   0.00000)\n",
      "Cycle   3: node 4 (   0.02018) joins species 6 (   0.02419)\n",
      "Cycle   2: node 8 (   0.32240) joins species 1 (   0.18728)\n",
      "Cycle   1: species 3 (   0.09050) joins species 2 (   0.14050)\n",
      "last cycle:\n",
      " node 3  (   0.01537) joins node 4  (   0.06925) joins node 8  (   0.02487)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 99:\n",
      "\n",
      "Cycle   5: species 8 (   0.28472) joins species 7 (   0.29414)\n",
      "Cycle   4: species 5 (   0.00001) joins species 4 (   0.00000)\n",
      "Cycle   3: species 6 (   0.01862) joins node 5 (   0.01737)\n",
      "Cycle   2: species 2 (   0.08624) joins species 3 (   0.08149)\n",
      "Cycle   1: node 8 (   0.27578) joins species 1 (   0.12495)\n",
      "last cycle:\n",
      " node 2  (   0.01545) joins node 6  (   0.05311) joins node 8  (   0.01148)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Data set # 100:\n",
      "\n",
      "Cycle   5: species 8 (   0.35513) joins species 7 (   0.38852)\n",
      "Cycle   4: node 8 (   0.30389) joins species 1 (   0.12034)\n",
      "Cycle   3: species 5 (   0.00001) joins species 4 (   0.00000)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cycle   2: node 5 (   0.02195) joins species 6 (   0.02937)\n",
      "Cycle   1: node 5 (   0.04619) joins species 3 (   0.11379)\n",
      "last cycle:\n",
      " species 2  (   0.14564) joins node 5  (   0.00520) joins node 8  (   0.03860)\n",
      "\n",
      "Output written on file \"outfile\"\n",
      "\n",
      "Tree written on file \"outtree\"\n",
      "\n",
      "Done.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "rm /data/user/home/jziad/info_302_class.git/SeqSearch/outtree\n",
    "/data/user/home/jziad/info_302_class.git/SeqSearch/phylip-3.697/exe/neighbor <<EOF\n",
    "M\n",
    "100\n",
    "3\n",
    "y\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "After you have created 100 trees. Run ```consense``` to get a consensus tree:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "engine": "bash",
    "eval": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[2J\u001b[H\n",
      "\u001b[2J\u001b[H\n",
      "Consensus tree program, version 3.697\n",
      "\n",
      "Settings for this run:\n",
      " C         Consensus type (MRe, strict, MR, Ml):  Majority rule (extended)\n",
      " O                                Outgroup root:  No, use as outgroup species  1\n",
      " R                Trees to be treated as Rooted:  No\n",
      " T           Terminal type (IBM PC, ANSI, none):  ANSI\n",
      " 1                Print out the sets of species:  Yes\n",
      " 2         Print indications of progress of run:  Yes\n",
      " 3                               Print out tree:  Yes\n",
      " 4               Write out trees onto tree file:  Yes\n",
      "\n",
      "Are these settings correct? (type Y or the letter for one to change)\n",
      "\n",
      "Consensus tree written to file \"outtree\"\n",
      "\n",
      "Output written to file \"outfile\"\n",
      "\n",
      "Done.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "rm outfile\n",
    "mv outtree intree\n",
    "consense <<EOF\n",
    "y\n",
    "EOF\n",
    "mv outtree neighbor_bootstrap.tre\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "# Maximum likelihood tree using RAxML\n",
    "\n",
    "RAxML is a very efficient maximum likelihood phylogenetic analysis software that can run on a HPC settings. The software is generally used for very large datasets.\n",
    "\n",
    "Download RAxML from https://github.com/stamatak/standard-RAxML. There are several way to compile RAxML. For use with our virtual machine we will use a very simple option."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "engine": "bash",
    "eval": false,
    "lines_to_next_cell": 0
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cloning into 'standard-RAxML'...\n",
      "remote: Enumerating objects: 2590, done.        \n",
      "remote: Total 2590 (delta 0), reused 0 (delta 0), pack-reused 2590        \n",
      "Receiving objects: 100% (2590/2590), 39.66 MiB | 55.94 MiB/s, done.\n",
      "Resolving deltas: 100% (1836/1836), done.\n",
      "rm -f *.o raxmlHPC\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o axml.o axml.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o optimizeModel.o optimizeModel.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o multiple.o multiple.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o searchAlgo.o searchAlgo.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o topologies.o topologies.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o parsePartitions.o parsePartitions.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o treeIO.o treeIO.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o models.o models.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o bipartitionList.o bipartitionList.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o rapidBootstrap.o rapidBootstrap.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o evaluatePartialGenericSpecial.o evaluatePartialGenericSpecial.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o evaluateGenericSpecial.o evaluateGenericSpecial.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o newviewGenericSpecial.o newviewGenericSpecial.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o makenewzGenericSpecial.o makenewzGenericSpecial.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o classify.o classify.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o fastDNAparsimony.o fastDNAparsimony.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o fastSearch.o fastSearch.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o leaveDropping.o leaveDropping.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o rmqs.o rmqs.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o rogueEPA.o rogueEPA.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o ancestralStates.o ancestralStates.c\n",
      "gcc  -D_GNU_SOURCE -fomit-frame-pointer -funroll-loops -O2 -msse    -c -o mem_alloc.o mem_alloc.c\n",
      "gcc  -c -o eigen.o eigen.c \n",
      "gcc  -o raxmlHPC axml.o  optimizeModel.o multiple.o searchAlgo.o topologies.o parsePartitions.o treeIO.o models.o bipartitionList.o rapidBootstrap.o evaluatePartialGenericSpecial.o evaluateGenericSpecial.o newviewGenericSpecial.o makenewzGenericSpecial.o   classify.o fastDNAparsimony.o fastSearch.o leaveDropping.o rmqs.o rogueEPA.o ancestralStates.o  mem_alloc.o  eigen.o -lm  \n"
     ]
    }
   ],
   "source": [
    "git clone https://github.com/stamatak/standard-RAxML.git\n",
    "cd standard-RAxML/\n",
    "make -f Makefile.gcc\n",
    "rm *.o\n",
    "cd .."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "RAxML requires specification of a model. For nucleotide sequence the best option GTRGAMMA. For protein sequence any of the JTT models are fine. They are expanded PAM matrices.\n",
    "\n",
    "But we can tell RAXML for find the best model for us."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "engine": "bash",
    "eval": false,
    "lines_to_next_cell": 0
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "\n",
      "\n",
      "IMPORTANT WARNING: Sequences NP_000537.3 and XP_001172077.2 are exactly identical\n",
      "\n",
      "IMPORTANT WARNING\n",
      "Found 1 sequence that is exactly identical to other sequences in the alignment.\n",
      "Normally they should be excluded from the analysis.\n",
      "\n",
      "Just in case you might need it, an alignment file with \n",
      "sequence duplicates removed is printed to file p53.afa.reduced\n",
      "\n",
      "\n",
      "This is RAxML version 8.2.12 released by Alexandros Stamatakis on May 2018.\n",
      "\n",
      "With greatly appreciated code contributions by:\n",
      "Andre Aberer      (HITS)\n",
      "Simon Berger      (HITS)\n",
      "Alexey Kozlov     (HITS)\n",
      "Kassian Kobert    (HITS)\n",
      "David Dao         (KIT and HITS)\n",
      "Sarah Lutteropp   (KIT and HITS)\n",
      "Nick Pattengale   (Sandia)\n",
      "Wayne Pfeiffer    (SDSC)\n",
      "Akifumi S. Tanabe (NRIFS)\n",
      "Charlie Taylor    (UF)\n",
      "\n",
      "\n",
      "Alignment has 275 distinct alignment patterns\n",
      "\n",
      "Proportion of gaps and completely undetermined characters in this alignment: 6.31%\n",
      "\n",
      "RAxML rapid hill-climbing mode\n",
      "\n",
      "Using 1 distinct models/data partitions with joint branch length optimization\n",
      "\n",
      "\n",
      "Executing 1 inferences on the original alignment using 1 distinct randomized MP trees\n",
      "\n",
      "All free model parameters will be estimated by RAxML\n",
      "GAMMA model of rate heterogeneity, ML estimate of alpha-parameter\n",
      "\n",
      "GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units\n",
      "\n",
      "Partition: 0\n",
      "Alignment Patterns: 275\n",
      "Name: No Name Provided\n",
      "DataType: AA\n",
      "Substitution Matrix: AUTO\n",
      "Using fixed base frequencies\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "RAxML was called as follows:\n",
      "\n",
      "./standard-RAxML/raxmlHPC -m PROTGAMMAAUTO -s p53.afa -p 12345 -n p53 \n",
      "\n",
      "\n",
      "Partition: 0 with name: No Name Provided\n",
      "Base frequencies: 0.087 0.044 0.039 0.057 0.019 0.037 0.058 0.083 0.024 0.048 0.086 0.062 0.020 0.038 0.046 0.070 0.061 0.014 0.035 0.071 \n",
      "\n",
      "Automatic protein model assignment algorithm using ML criterion:\n",
      "\n",
      "\tPartition: 0 best-scoring AA model: JTTDCMUT likelihood -3410.765493 with empirical base frequencies\n",
      "\n",
      "\n",
      "Automatic protein model assignment algorithm using ML criterion:\n",
      "\n",
      "\tPartition: 0 best-scoring AA model: JTTDCMUT likelihood -3404.812349 with empirical base frequencies\n",
      "\n",
      "\n",
      "Automatic protein model assignment algorithm using ML criterion:\n",
      "\n",
      "\tPartition: 0 best-scoring AA model: JTTDCMUT likelihood -3404.762153 with empirical base frequencies\n",
      "\n",
      "\n",
      "Automatic protein model assignment algorithm using ML criterion:\n",
      "\n",
      "\tPartition: 0 best-scoring AA model: JTTDCMUT likelihood -3404.761664 with empirical base frequencies\n",
      "\n",
      "\n",
      "Inference[0]: Time 16.188066 GAMMA-based likelihood -3404.761659, best rearrangement setting 5\n",
      "\n",
      "\n",
      "Conducting final model optimizations on all 1 trees under GAMMA-based models ....\n",
      "\n",
      "Automatic protein model assignment algorithm using ML criterion:\n",
      "\n",
      "\tPartition: 0 best-scoring AA model: JTTDCMUT likelihood -3404.761659 with empirical base frequencies\n",
      "\n",
      "\n",
      "Inference[0] final GAMMA-based Likelihood: -3404.761659 tree written to file /data/user/home/jziad/info_302_class.git/SeqSearch/RAxML_result.p53\n",
      "\n",
      "\n",
      "Starting final GAMMA-based thorough Optimization on tree 0 likelihood -3404.761659 .... \n",
      "\n",
      "Final GAMMA-based Score of best tree -3404.761659\n",
      "\n",
      "Program execution info written to /data/user/home/jziad/info_302_class.git/SeqSearch/RAxML_info.p53\n",
      "Best-scoring ML tree written to: /data/user/home/jziad/info_302_class.git/SeqSearch/RAxML_bestTree.p53\n",
      "\n",
      "Overall execution time: 20.661408 secs or 0.005739 hours or 0.000239 days\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#cd 05-MolEvol\n",
    "./standard-RAxML/raxmlHPC -m PROTGAMMAAUTO -s p53.afa -p 12345 -n p53"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This will create the tree as ```RAxML_bestTree.p53```. \n",
    "\n",
    "We can also run a bootstrap analysis:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "((NP_001040616.1:0.02634177446863036898,(NP_000537.3:0.00000100000050002909,XP_001172077.2:0.00000100000050002909):0.01791132739213887812):0.05778258529111870295,(NP_001003210.1:0.11067073856982224955,((NP_001001903.1:0.57204194774946504065,NP_001258749.1:0.55676768397545683964):0.58753189363785462973,NP_776626.1:0.10465936615772329643):0.03386727151490174365):0.03136581737618214516,NP_112251.2:0.21548755322834986514):0.0;\n"
     ]
    }
   ],
   "source": [
    "cat RAxML_bestTree.p53"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "engine": "bash",
    "eval": false,
    "lines_to_next_cell": 0
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "RAxML can't, parse the alignment file as phylip file \n",
      "it will now try to parse it as FASTA file\n",
      "\n",
      "\n",
      "\n",
      "IMPORTANT WARNING: Sequences NP_000537.3 and XP_001172077.2 are exactly identical\n",
      "\n",
      "IMPORTANT WARNING\n",
      "Found 1 sequence that is exactly identical to other sequences in the alignment.\n",
      "Normally they should be excluded from the analysis.\n",
      "\n",
      "An alignment file with sequence duplicates removed has already\n",
      "been printed to file p53.afa.reduced\n",
      "\n",
      "\n",
      "This is RAxML version 8.2.12 released by Alexandros Stamatakis on May 2018.\n",
      "\n",
      "With greatly appreciated code contributions by:\n",
      "Andre Aberer      (HITS)\n",
      "Simon Berger      (HITS)\n",
      "Alexey Kozlov     (HITS)\n",
      "Kassian Kobert    (HITS)\n",
      "David Dao         (KIT and HITS)\n",
      "Sarah Lutteropp   (KIT and HITS)\n",
      "Nick Pattengale   (Sandia)\n",
      "Wayne Pfeiffer    (SDSC)\n",
      "Akifumi S. Tanabe (NRIFS)\n",
      "Charlie Taylor    (UF)\n",
      "\n",
      "\n",
      "Alignment has 275 distinct alignment patterns\n",
      "\n",
      "Proportion of gaps and completely undetermined characters in this alignment: 6.31%\n",
      "\n",
      "RAxML rapid bootstrapping and subsequent ML search\n",
      "\n",
      "Using 1 distinct models/data partitions with joint branch length optimization\n",
      "\n",
      "\n",
      "\n",
      "Executing 10 rapid bootstrap inferences and thereafter a thorough ML search \n",
      "\n",
      "All free model parameters will be estimated by RAxML\n",
      "GAMMA model of rate heterogeneity, ML estimate of alpha-parameter\n",
      "\n",
      "GAMMA Model parameters will be estimated up to an accuracy of 0.1000000000 Log Likelihood units\n",
      "\n",
      "Partition: 0\n",
      "Alignment Patterns: 275\n",
      "Name: No Name Provided\n",
      "DataType: AA\n",
      "Substitution Matrix: JTT\n",
      "Using fixed base frequencies\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "RAxML was called as follows:\n",
      "\n",
      "standard-RAxML/raxmlHPC -m PROTGAMMAJTT -s p53.afa -p 12345 -# 10 -f a -n p53_c -x 12345 \n",
      "\n",
      "\n",
      "\n",
      "Time for BS model parameter optimization 0.189202\n",
      "Bootstrap[0]: Time 1.568508 seconds, bootstrap likelihood -3444.621750, best rearrangement setting 13\n",
      "Bootstrap[1]: Time 1.271221 seconds, bootstrap likelihood -3397.777681, best rearrangement setting 13\n",
      "Bootstrap[2]: Time 1.096089 seconds, bootstrap likelihood -3606.894140, best rearrangement setting 12\n",
      "Bootstrap[3]: Time 1.111096 seconds, bootstrap likelihood -3333.699144, best rearrangement setting 15\n",
      "Bootstrap[4]: Time 1.079804 seconds, bootstrap likelihood -3440.030606, best rearrangement setting 10\n",
      "Bootstrap[5]: Time 1.043689 seconds, bootstrap likelihood -3368.346682, best rearrangement setting 10\n",
      "Bootstrap[6]: Time 1.108973 seconds, bootstrap likelihood -3497.137063, best rearrangement setting 6\n",
      "Bootstrap[7]: Time 0.997236 seconds, bootstrap likelihood -3299.336964, best rearrangement setting 14\n",
      "Bootstrap[8]: Time 0.981990 seconds, bootstrap likelihood -3310.629653, best rearrangement setting 5\n",
      "Bootstrap[9]: Time 1.279483 seconds, bootstrap likelihood -3515.377538, best rearrangement setting 7\n",
      "\n",
      "\n",
      "Overall Time for 10 Rapid Bootstraps 11.552607 seconds\n",
      "Average Time per Rapid Bootstrap 1.155261 seconds\n",
      "\n",
      "Starting ML Search ...\n",
      "\n",
      "Fast ML optimization finished\n",
      "\n",
      "Fast ML search Time: 4.577786 seconds\n",
      "\n",
      "Slow ML Search 0 Likelihood: -3444.074120\n",
      "Slow ML Search 1 Likelihood: -3444.074120\n",
      "Slow ML optimization finished\n",
      "\n",
      "Slow ML search Time: 3.140415 seconds\n",
      "Thorough ML search Time: 1.518673 seconds\n",
      "\n",
      "Final ML Optimization Likelihood: -3444.074120\n",
      "\n",
      "Model Information:\n",
      "\n",
      "Model Parameters of Partition 0, Name: No Name Provided, Type of Data: AA\n",
      "alpha: 0.610375\n",
      "Tree-Length: 2.315707\n",
      "rate A <-> R: 0.603538\n",
      "rate A <-> N: 0.561915\n",
      "rate A <-> D: 0.842872\n",
      "rate A <-> C: 0.582726\n",
      "rate A <-> Q: 0.593132\n",
      "rate A <-> E: 1.092612\n",
      "rate A <-> G: 1.862643\n",
      "rate A <-> H: 0.280957\n",
      "rate A <-> I: 0.374610\n",
      "rate A <-> L: 0.312175\n",
      "rate A <-> K: 0.364204\n",
      "rate A <-> M: 0.561915\n",
      "rate A <-> F: 0.156087\n",
      "rate A <-> P: 2.018730\n",
      "rate A <-> S: 3.933403\n",
      "rate A <-> T: 4.942768\n",
      "rate A <-> W: 0.093652\n",
      "rate A <-> Y: 0.114464\n",
      "rate A <-> V: 3.100937\n",
      "rate R <-> N: 0.468262\n",
      "rate R <-> D: 0.166493\n",
      "rate R <-> C: 1.175858\n",
      "rate R <-> Q: 3.225806\n",
      "rate R <-> E: 0.301769\n",
      "rate R <-> G: 1.425598\n",
      "rate R <-> H: 3.413111\n",
      "rate R <-> I: 0.228928\n",
      "rate R <-> L: 0.395421\n",
      "rate R <-> K: 6.722164\n",
      "rate R <-> M: 0.457856\n",
      "rate R <-> F: 0.052029\n",
      "rate R <-> P: 0.770031\n",
      "rate R <-> S: 1.050989\n",
      "rate R <-> T: 0.665973\n",
      "rate R <-> W: 1.311134\n",
      "rate R <-> Y: 0.208117\n",
      "rate R <-> V: 0.176899\n",
      "rate N <-> D: 5.494277\n",
      "rate N <-> C: 0.353798\n",
      "rate N <-> Q: 0.894901\n",
      "rate N <-> E: 0.603538\n",
      "rate N <-> G: 0.842872\n",
      "rate N <-> H: 4.068678\n",
      "rate N <-> I: 0.489074\n",
      "rate N <-> L: 0.124870\n",
      "rate N <-> K: 2.736733\n",
      "rate N <-> M: 0.312175\n",
      "rate N <-> F: 0.104058\n",
      "rate N <-> P: 0.156087\n",
      "rate N <-> S: 5.234131\n",
      "rate N <-> T: 2.414152\n",
      "rate N <-> W: 0.083247\n",
      "rate N <-> Y: 0.728408\n",
      "rate N <-> V: 0.166493\n",
      "rate D <-> C: 0.104058\n",
      "rate D <-> Q: 0.509886\n",
      "rate D <-> E: 7.981270\n",
      "rate D <-> G: 1.352758\n",
      "rate D <-> H: 1.165453\n",
      "rate D <-> I: 0.114464\n",
      "rate D <-> L: 0.072841\n",
      "rate D <-> K: 0.270552\n",
      "rate D <-> M: 0.156087\n",
      "rate D <-> F: 0.041623\n",
      "rate D <-> P: 0.156087\n",
      "rate D <-> S: 0.613944\n",
      "rate D <-> T: 0.395421\n",
      "rate D <-> W: 0.041623\n",
      "rate D <-> Y: 0.478668\n",
      "rate D <-> V: 0.322581\n",
      "rate C <-> Q: 0.093652\n",
      "rate C <-> E: 0.052029\n",
      "rate C <-> G: 0.613944\n",
      "rate C <-> H: 0.718002\n",
      "rate C <-> I: 0.176899\n",
      "rate C <-> L: 0.239334\n",
      "rate C <-> K: 0.072841\n",
      "rate C <-> M: 0.322581\n",
      "rate C <-> F: 0.811655\n",
      "rate C <-> P: 0.145682\n",
      "rate C <-> S: 2.320499\n",
      "rate C <-> T: 0.437045\n",
      "rate C <-> W: 1.196670\n",
      "rate C <-> Y: 2.174818\n",
      "rate C <-> V: 0.645161\n",
      "rate Q <-> E: 3.361082\n",
      "rate Q <-> G: 0.270552\n",
      "rate Q <-> H: 6.212279\n",
      "rate Q <-> I: 0.093652\n",
      "rate Q <-> L: 0.749220\n",
      "rate Q <-> K: 3.038502\n",
      "rate Q <-> M: 0.447451\n",
      "rate Q <-> F: 0.041623\n",
      "rate Q <-> P: 1.706556\n",
      "rate Q <-> S: 0.551509\n",
      "rate Q <-> T: 0.530697\n",
      "rate Q <-> W: 0.187305\n",
      "rate Q <-> Y: 0.249740\n",
      "rate Q <-> V: 0.208117\n",
      "rate E <-> G: 1.238293\n",
      "rate E <-> H: 0.270552\n",
      "rate E <-> I: 0.124870\n",
      "rate E <-> L: 0.093652\n",
      "rate E <-> K: 1.883455\n",
      "rate E <-> M: 0.187305\n",
      "rate E <-> F: 0.052029\n",
      "rate E <-> P: 0.187305\n",
      "rate E <-> S: 0.312175\n",
      "rate E <-> T: 0.332986\n",
      "rate E <-> W: 0.104058\n",
      "rate E <-> Y: 0.072841\n",
      "rate E <-> V: 0.468262\n",
      "rate G <-> H: 0.239334\n",
      "rate G <-> I: 0.062435\n",
      "rate G <-> L: 0.062435\n",
      "rate G <-> K: 0.280957\n",
      "rate G <-> M: 0.145682\n",
      "rate G <-> F: 0.052029\n",
      "rate G <-> P: 0.249740\n",
      "rate G <-> S: 2.091571\n",
      "rate G <-> T: 0.343392\n",
      "rate G <-> W: 0.572320\n",
      "rate G <-> Y: 0.083247\n",
      "rate G <-> V: 0.489074\n",
      "rate H <-> I: 0.166493\n",
      "rate H <-> L: 0.582726\n",
      "rate H <-> K: 0.468262\n",
      "rate H <-> M: 0.343392\n",
      "rate H <-> F: 0.416233\n",
      "rate H <-> P: 1.196670\n",
      "rate H <-> S: 0.759625\n",
      "rate H <-> T: 0.478668\n",
      "rate H <-> W: 0.083247\n",
      "rate H <-> Y: 5.962539\n",
      "rate H <-> V: 0.114464\n",
      "rate I <-> L: 2.382934\n",
      "rate I <-> K: 0.218522\n",
      "rate I <-> M: 4.984391\n",
      "rate I <-> F: 0.926119\n",
      "rate I <-> P: 0.104058\n",
      "rate I <-> S: 0.416233\n",
      "rate I <-> T: 2.549428\n",
      "rate I <-> W: 0.093652\n",
      "rate I <-> Y: 0.332986\n",
      "rate I <-> V: 10.000000\n",
      "rate L <-> K: 0.145682\n",
      "rate L <-> M: 4.037461\n",
      "rate L <-> F: 2.580645\n",
      "rate L <-> P: 1.061394\n",
      "rate L <-> S: 0.613944\n",
      "rate L <-> T: 0.260146\n",
      "rate L <-> W: 0.541103\n",
      "rate L <-> Y: 0.249740\n",
      "rate L <-> V: 1.873049\n",
      "rate K <-> M: 0.676379\n",
      "rate K <-> F: 0.041623\n",
      "rate K <-> P: 0.218522\n",
      "rate K <-> S: 0.489074\n",
      "rate K <-> T: 1.071800\n",
      "rate K <-> W: 0.104058\n",
      "rate K <-> Y: 0.083247\n",
      "rate K <-> V: 0.145682\n",
      "rate M <-> F: 0.447451\n",
      "rate M <-> P: 0.166493\n",
      "rate M <-> S: 0.301769\n",
      "rate M <-> T: 2.351717\n",
      "rate M <-> W: 0.249740\n",
      "rate M <-> Y: 0.187305\n",
      "rate M <-> V: 3.361082\n",
      "rate F <-> P: 0.176899\n",
      "rate F <-> S: 0.957336\n",
      "rate F <-> T: 0.124870\n",
      "rate F <-> W: 0.551509\n",
      "rate F <-> Y: 5.577523\n",
      "rate F <-> V: 0.645161\n",
      "rate P <-> S: 2.965661\n",
      "rate P <-> T: 1.227888\n",
      "rate P <-> W: 0.062435\n",
      "rate P <-> Y: 0.104058\n",
      "rate P <-> V: 0.239334\n",
      "rate S <-> T: 4.963580\n",
      "rate S <-> W: 0.364204\n",
      "rate S <-> Y: 0.655567\n",
      "rate S <-> V: 0.395421\n",
      "rate T <-> W: 0.124870\n",
      "rate T <-> Y: 0.218522\n",
      "rate T <-> V: 1.165453\n",
      "rate W <-> Y: 0.738814\n",
      "rate W <-> V: 0.260146\n",
      "rate Y <-> V: 0.166493\n",
      "\n",
      "freq pi(A): 0.076748\n",
      "freq pi(R): 0.051691\n",
      "freq pi(N): 0.042645\n",
      "freq pi(D): 0.051544\n",
      "freq pi(C): 0.019803\n",
      "freq pi(Q): 0.040752\n",
      "freq pi(E): 0.061830\n",
      "freq pi(G): 0.073152\n",
      "freq pi(H): 0.022944\n",
      "freq pi(I): 0.053761\n",
      "freq pi(L): 0.091904\n",
      "freq pi(K): 0.058676\n",
      "freq pi(M): 0.023826\n",
      "freq pi(F): 0.040126\n",
      "freq pi(P): 0.050901\n",
      "freq pi(S): 0.068765\n",
      "freq pi(T): 0.058565\n",
      "freq pi(W): 0.014261\n",
      "freq pi(Y): 0.032102\n",
      "freq pi(V): 0.066004\n",
      "\n",
      "\n",
      "ML search took 9.307838 secs or 0.002586 hours\n",
      "\n",
      "Combined Bootstrap and ML search took 20.860485 secs or 0.005795 hours\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Drawing Bootstrap Support Values on best-scoring ML tree ...\n",
      "\n",
      "\n",
      "\n",
      "Found 1 tree in File /data/user/home/jziad/info_302_class.git/SeqSearch/RAxML_bestTree.p53_c\n",
      "\n",
      "\n",
      "\n",
      "Found 1 tree in File /data/user/home/jziad/info_302_class.git/SeqSearch/RAxML_bestTree.p53_c\n",
      "\n",
      "Program execution info written to /data/user/home/jziad/info_302_class.git/SeqSearch/RAxML_info.p53_c\n",
      "All 10 bootstrapped trees written to: /data/user/home/jziad/info_302_class.git/SeqSearch/RAxML_bootstrap.p53_c\n",
      "\n",
      "Best-scoring ML tree written to: /data/user/home/jziad/info_302_class.git/SeqSearch/RAxML_bestTree.p53_c\n",
      "\n",
      "Best-scoring ML tree with support values written to: /data/user/home/jziad/info_302_class.git/SeqSearch/RAxML_bipartitions.p53_c\n",
      "\n",
      "Best-scoring ML tree with support values as branch labels written to: /data/user/home/jziad/info_302_class.git/SeqSearch/RAxML_bipartitionsBranchLabels.p53_c\n",
      "\n",
      "Overall execution time for full ML analysis: 20.892486 secs or 0.005803 hours or 0.000242 days\n",
      "\n"
     ]
    }
   ],
   "source": [
    "standard-RAxML/raxmlHPC -m PROTGAMMAJTT -s p53.afa -p 12345 -# 10 -f a -n p53_c -x 12345"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Unlike ```consense``` in ```phylip```. RAxML already geneartes a tree will bootstrap values already included. This is the file ```RAxML_bipartitions.p53_c``` How nice!\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "jupytext": {
   "cell_metadata_filter": "eval,engine,-all",
   "main_language": "R",
   "notebook_metadata_filter": "-all"
  },
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
